---
title: "Manuscript_1"
author: Deborah GÃ©rard^[University of Luxembourg - FSTM - DLSM - Systems Biology group - Epigenetics team]
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  html_document:
    keep_md: true
    df_print: paged
    toc: true
    toc_depth: 6
  pdf_document: default
editor_options:
  chunk_output_type: console
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = TRUE, 
                      warning = FALSE, 
                      message = FALSE, 
                      results = "as.is", 
                      fig.height = 3, 
                      cache = FALSE, 
                      tidy = TRUE, 
                      fig.pos = "H")
```

#### *1. Create a Singularity container*

Work within a Singularity container to ensure reproducibility

```{bash Sing_crea, eval = FALSE, echo = TRUE}
# Initiate virtual box Vagrant
cd $HOME/Manuscript_1_vagrant

vagrant up --provision
# vagrant reload
vagrant ssh

# Check the number of allocated cores and memory
free -h
nproc

##
#rsync -avzPhu iris-cluster:/home/users/dgerard/Singularity_containers/Manuscript_1_singularity.sif ~/Downloads/

#scp -r -P 2222 -i /Users/deborah.gerard/singularity-vm/.vagrant/machines/default/virtualbox/private_key \
#~/Downloads/Manuscript_1_singularity.sif vagrant@127.0.0.1:/home/vagrant/Manuscript_1_singularity/

##
# Check singularity version
singularity version   # 3.9.0

# Prepare the singularity definition file
mkdir Manuscript_1_singularity && cd Manuscript_1_singularity
touch Manuscript_1_singularity.def

# Build the temporary container based on Ubuntu 20.04 from Docker for development 
sudo singularity build --sandbox Manuscript_1_singularity_tmp Manuscript_1_singularity.def

# Run the container in writable mode to make changes
sudo singularity shell --writable Manuscript_1_singularity_tmp

# TEST
sudo singularity build --sandbox Manuscript_1_singularity_tmp Manuscript_1_singularity.sif

# Now that all necessay packages and library have been installed, build the container
sudo singularity build Manuscript_1_singularity.sif Manuscript_1_singularity_tmp

# Start R
singularity exec Manuscript_1_singularity.sif R

# Check R version
R.Version()
```

**Note**: R version is 4.2.3 (2023-03-15)

##### *1.1 Customize the Singularity container*

Install different R libraries

```{r cust, eval = FALSE, echo = TRUE}
# tidyverse, ggpubr, rstatix
install.packages(c("tidyverse", 
                   "ggpubr", 
                   "rstatix"))
# plotgardener
BiocManager::install("plotgardener")

# BSgenome.Hsapiens.UCSC.hg38
BiocManager::install("BSgenome.Hsapiens.UCSC.hg38")

# extrafont
install.packages("extrafont")

# showtext
install.packages("showtext")

# TxDb.Hsapiens.UCSC.hg38.knownGene
BiocManager::install("TxDb.Hsapiens.UCSC.hg38.knownGene")

# org.Hs.eg.db
BiocManager::install("org.Hs.eg.db")

# JASPAR2020
BiocManager::install("JASPAR2020")

# TFBSTools
BiocManager::install("TFBSTools")

# ggseqlogo
install.packages("ggseqlogo")

# TxDb.Hsapiens.UCSC.hg38.refGene
BiocManager::install("TxDb.Hsapiens.UCSC.hg38.refGene")

# LDlinkR
install.packages("LDlinkR")

# AllelicImbalance
BiocManager::install("AllelicImbalance")

# ggflowchart
install.packages("ggflowchart")

# Issue between dplyr, BiocFilrCache and biomaRt - > downgrade dbplyr
install.packages("devtools")
devtools::install_version("dbplyr", version = "2.3.4")

# ggmanh
BiocManager::install("ggmanh")
```

Load libraries and check their versions

```{r load_lib, eval = FALSE, echo = TRUE}
# Load
library("tidyverse")
library("rstatix")
library("ggpubr")
library("plotgardener")
#library("BSgenome.Hsapiens.UCSC.hg38")
library("TxDb.Hsapiens.UCSC.hg38.knownGene")
library("TxDb.Hsapiens.UCSC.hg38.refGene")
library("org.Hs.eg.db")
library("extrafont")
library("showtext")
library("JASPAR2020")
library("TFBSTools")
library("ggseqlogo")
library("biomaRt")
library("LDlinkR")
library("AllelicImbalance")
library("png")
library("AnnotationDbi")
library("scales")
library("flowCore")
library("ggcyto")
library("ggmanh")
library("rtracklayer")
library("plyranges")
library("RColorBrewer")

# Check
packageVersion("tidyverse") #2.0.0
packageVersion("rstatix") #0.7.2
packageVersion("ggpubr")  #0.6.0
packageVersion("plotgardener")  #1.4.2
#packageVersion("BSgenome.Hsapiens.UCSC.hg38") #1.4.5
#packageVersion("TxDb.Hsapiens.UCSC.hg38.knownGene") #3.16.0
packageVersion("TxDb.Hsapiens.UCSC.hg38.refGene") #3.15.0
packageVersion("org.Hs.eg.db")  #3.16.0
packageVersion("extrafont") #0.19
packageVersion("showtext")  #0.9.7
packageVersion("JASPAR2020")  #0.99.10
packageVersion("TFBSTools") #1.36.0
packageVersion("ggseqlogo") #0.2
packageVersion("biomaRt") #2.54.1
packageVersion("LDlinkR") #1.4.0
packageVersion("AllelicImbalance")  #1.36.0
packageVersion("png") #0.1.8
packageVersion("AnnotationDbi") #1.60.2
#packageVersion("dbplyr")  #2.5.0
packageVersion("scales")  #1.3.0
packageVersion("flowCore")  #2.10.0
packageVersion("ggcyto")  #1.26.4
packageVersion("ggmanh")  #1.2.0
packageVersion("rtracklayer") #1.58.0
packageVersion("plyranges") #1.18.0
packageVersion("RColorBrewer")  #1.1.3

# Import all available fonts
font_import()
```

# FIGURES
Generating the manhattan plot for the PD GWAS is memory demanding. Go to the HPC
```{bash, eval = FALSE, echo = TRUE}
#ssh iris-cluster
#salloc --nodes=1 -p interactive --qos debug -C batch -t 02:00:00 --mem 24G

# Load the Singularity module
#module load tools/Singularity/3.8.1
#singularity shell --bind /scratch/users/dgerard:/scratch/users/dgerard $HOME/Singularity_containers/Manuscript_1_singularity.sif
```

### FIGURE 1
```{r fig1, eval = FALSE, echo = TRUE}
########## FIGURE 1 ##########
##############################
# Save as TIFF, 300 ppi
tiff("/home/vagrant/Manuscript_1/FIGURE1/FIGURE1.tiff",
     width = 8.27,
     height = 9.5,
     units = "in",
     res = 300,
     compression = "lzw")

# Create a A4 blank page
pageCreate(width = 8.27, 
           height = 9.5, 
           default.units = "inches",
           showGuides = TRUE)

#### PANEL A - 
# text Figure 1
plotText(label = "Figure 1", 
         fontsize = 14,
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 0.25, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# text A
plotText(label = "A", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 0.5, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

##########################################
#### Figure 1A Data generation scheme ####
##########################################
# Figure 1A generated in Biorender
fig1A = readPNG("/home/vagrant/Manuscript_1/FIGURE1/Fig1A_Biorender_vertical.png")

plotRaster(image = fig1A,
           x = 0.5, 
           y = 2.5, 
           default.units = "inches",
           width = 2.5, 
           height = 3.75, 
           just = "left",
           interpolate = TRUE)

##########################################
#### Figure 1B Odd ratio #################
##########################################
# text B
plotText(label = "B", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 3.0, 
         y = 0.5, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# Load data obtained from Jochen
RNAseq_odd = read_delim("/home/vagrant/Manuscript_1/FIGURE1/REFORMAT_tpm_gene_sets.gsa.txt",
                        delim = "\t",
                        col_names = TRUE)

# Calculate odd ratio
RNAseq_odd = RNAseq_odd %>% 
  dplyr::filter(!VARIABLE %in% c("microglia")) %>% 
  mutate(odd_R = exp(BETA)) %>% 
  mutate(VARIABLE = factor(VARIABLE, 
                           levels = c("smNPC",
                                      "D15negsort",
                                      "D15possort",
                                      "D30postsort",
                                      "D50negsort",
                                      "D50possort",
                                      "astrocytes")))

# Plot
fig1B = ggplot(RNAseq_odd,
       aes(x = odd_R,
           y = VARIABLE)) +
  geom_point(shape = 16, size = 3) +
  geom_errorbarh(aes(xmin = odd_R - SE, 
                     xmax = odd_R + SE), 
                 height = 0.25) +
  geom_vline(xintercept = 1.0, linetype = "dashed", color = "grey") +
  theme_classic() +
  theme(axis.title = element_text(face = "bold"),
        axis.text = element_text(face = "bold")) +
  scale_y_discrete(labels = c("smNPC" = "smNPC", 
                              "D15negsort" = "non-mDAN D15", 
                              "D15possort" = "mDAN D15", 
                              "D30postsort" = "mDAN D30", 
                              "D50negsort" = "non-mDAN D50",
                              "D50possort" = "mDAN D50",
                              "astrocytes" = "Astrocytes")) +
  geom_text(label = round(RNAseq_odd$P, 
                          digits = 3),
            nudge_y = 0.40) +
  xlab("Odds ratio \n(95% Confidence Interval)") +
  ylab("")

# Place the plot
plotGG(plot = fig1B, 
       x = 3.25, 
       y = 0.5, 
       width = 4.5, 
       height = 4.5,
       just = c("left", "top"), 
       default.units = "inches")

####################################
#### Figure 1C LowC - Chr4 #########
####################################
# text C
plotText(label = "C", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 5.25, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# Load TH REP1 mCHERRY mDAN neurons D30 (N1) at 50kb resolution
mDAN.D30.50kb = readHic(file = "/home/vagrant/epifunc/LowC_smNPC_THpos_neur/FANC_output_N1_mDAN_D30/hic/juicer_like/N1_mDAN_D30.juicer.50kb.hic",
                     chrom = "chr4",
                     assembly = "hg38",
                     resolution = 50000,
                     res_scale = "BP",
                     norm = "NONE",
                     matrix = "observed")

# Load TH REP1 mCHERRY mDAN neurons D30 (N1) at 25kb resolution
mDAN.D30.25kb = readHic(file = "/home/vagrant/epifunc/LowC_smNPC_THpos_neur/FANC_output_N1_mDAN_D30/hic/juicer_like/N1_mDAN_D30.juicer.25kb.hic",
                     chrom = "chr4",
                     assembly = "hg38",
                     resolution = 25000,
                     res_scale = "BP",
                     norm = "NONE",
                     matrix = "observed")

# Load TH REP1 mCHERRY smNPCs (N1) at 50kb resolution
smNPC.50kb = readHic(file = "/home/vagrant/epifunc/LowC_smNPC_THpos_neur/FANC_output_N1_smNPC/hic/juicer_like/N1_smNPC.juicer.50kb.hic",
                     chrom = "chr4",
                     assembly = "hg38",
                     resolution = 50000,
                     res_scale = "BP",
                     norm = "NONE",
                     matrix = "observed")

# Full chr4 to display
region.p.chr4 = pgParams(chrom = "chr4",
                         assembly = assembly(Genome = "hg38refGene",
                                             TxDb = "TxDb.Hsapiens.UCSC.hg38.refGene",
                                             OrgDb = "org.Hs.eg.db"),
                         just = c("left", "top"),
                         width = 3.5,
                         fontcolor = "black",
                         fill = "black")

# Plot full chr4 - smNPC on top of the diagonale and mDAN D30 at the bottom of the diagonale
LowC_smNPC.top = plotHicSquare(data = smNPC.50kb, 
                               params = region.p.chr4,
                               zrange = c(0, 5), 
                               resolution = "auto",
                               half = "top",
                               x = 0.5,
                               y = 5.5, 
                               height = 3.5)

annoHeatmapLegend(plot = LowC_smNPC.top, 
                  fontsize = 7,
                  x = 0.375, 
                  y = 6.0,
                  width = 0.07, 
                  height = 0.5, 
                  just = c("left", "top"),
                  default.units = "inches")

# smNPC
plotText(label = "smNPC", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 0.5625, 
         y = 5.625, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

LowC_mDAND30.bottom = plotHicSquare(data = mDAN.D30.50kb, 
                               params = region.p.chr4,
                               zrange = c(0, 5), 
                               resolution = "auto",
                               half = "bottom",
                               x = 0.5,
                               y = 5.5, 
                               height = 3.5)

annoHeatmapLegend(plot = LowC_mDAND30.bottom, fontsize = 7,
                  x = 4.0625, 
                  y = 6.0,
                  width = 0.07, 
                  height = 0.5,
                  just = c("left", "top"),
                  default.units = "inches")

# mDAN D30
plotText(label = "mDAN D30", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 3.0, 
         y = 8.9375, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# Add genome label of chr4
annoGenomeLabel(plot = LowC_mDAND30.bottom, 
                scale = "Mb", 
                axis = "x",
                x = 0.5, 
                y = 9.0625, 
                just = c("left", "top"))

# Annotate SNCA domain that will be zoomed in fid1D
SNCA.TAD = GRanges("chr4",
                   ranges = IRanges(start = 89000000,
                                    end = 91000000))

domainAnno = annoDomains(plot = LowC_mDAND30.bottom, 
                         data = SNCA.TAD,
                         half = "bottom", 
                         linecolor = "red")

###################################
#### Figure 1D SNCA locus #########
###################################
# Zoom on SNCA an add ATACseq data of the locus
# Define parameters for a small part of chr4 where SNCA is
# Add ATACseq track of smNPC, mDAN D15, mDAN D30, mDAN D50
# text D
plotText(label = "D", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 4.125, 
         y = 5.25, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

bw.path = "/home/vagrant/epifunc/"

# smNPC N1
bw.SNCA.smNPC.1 = readBigwig(file = paste0(bw.path, "BIGWIG/smNPC_I.bw"),
    chrom = "chr4", chromstart = 89000000, chromend = 91000000)

# smNPC N2
bw.SNCA.smNPC.2 = readBigwig(file = paste0(bw.path, "BIGWIG/smNPC_II.bw"),
    chrom = "chr4", chromstart = 89000000, chromend = 91000000)

# smNPC N3
bw.SNCA.smNPC.3 = readBigwig(file = paste0(bw.path, "BIGWIG/smNPC_III.bw"),
    chrom = "chr4", chromstart = 89000000, chromend = 91000000)

# mDAN D15 N1
bw.SNCA.mDAN.D15.1 = readBigwig(file = paste0(bw.path, "BIGWIG/D15_POS_I.bw"),
    chrom = "chr4", chromstart = 89000000, chromend = 91000000)

# mDAN D15 N2
bw.SNCA.mDAN.D15.2 = readBigwig(file = paste0(bw.path, "BIGWIG/D15_POS_II.bw"),
    chrom = "chr4", chromstart = 89000000, chromend = 91000000)

# mDAN D15 N3
bw.SNCA.mDAN.D15.3 = readBigwig(file = paste0(bw.path, "BIGWIG/D15_POS_III.bw"),
    chrom = "chr4", chromstart = 89000000, chromend = 91000000)

# mDAN D30 N1
bw.SNCA.mDAN.D30.1 = readBigwig(file = paste0(bw.path, "BIGWIG/D30_POS_I.bw"),
    chrom = "chr4", chromstart = 89000000, chromend = 91000000)

# mDAN D30 N2
bw.SNCA.mDAN.D30.2 = readBigwig(file = paste0(bw.path, "BIGWIG/D30_POS_II.bw"),
    chrom = "chr4", chromstart = 89000000, chromend = 91000000)

# mDAN D30 N3
bw.SNCA.mDAN.D30.3 = readBigwig(file = paste0(bw.path, "BIGWIG/D30_POS_III.bw"),
    chrom = "chr4", chromstart = 89000000, chromend = 91000000)

# mDAN D50 N1
bw.SNCA.mDAN.D50.1 = readBigwig(file = paste0(bw.path, "BIGWIG/HFFTHmCherry_D50_possort_S1.bw"),
    chrom = "chr4", chromstart = 89000000, chromend = 91000000)

# mDAN D50 N2
bw.SNCA.mDAN.D50.2 = readBigwig(file = paste0(bw.path, "BIGWIG/HFFTHmCherry_D50_possort_S3.bw"),
    chrom = "chr4", chromstart = 89000000, chromend = 91000000)

# mDAN D50 N3
bw.SNCA.mDAN.D50.3 = readBigwig(file = paste0(bw.path, "BIGWIG/HFFTHmCherry_D50_possort_S4.bw"),
    chrom = "chr4", chromstart = 89000000, chromend = 91000000)

# Add a scale next to the bigwig files - check the maximum to choose
scale.SNCA.max = max(c(bw.SNCA.smNPC.1$score,
                       bw.SNCA.smNPC.2$score,
                       bw.SNCA.smNPC.3$score,
                       bw.SNCA.mDAN.D15.1$score,
                       bw.SNCA.mDAN.D15.2$score,
                       bw.SNCA.mDAN.D15.3$score,
                       bw.SNCA.mDAN.D30.1$score,
                       bw.SNCA.mDAN.D30.2$score,
                       bw.SNCA.mDAN.D30.3$score,
                       bw.SNCA.mDAN.D50.1$score,
                       bw.SNCA.mDAN.D50.2$score,
                       bw.SNCA.mDAN.D50.3$score))

print(scale.SNCA.max)

# Define a small region of chr4
region.p.chr4.small = pgParams(chrom = "chr4",
                         chromstart = 89000000,
                         chromend = 91000000,
                         assembly = assembly(Genome = "hg38refGene",
                                               TxDb = "TxDb.Hsapiens.UCSC.hg38.refGene",
                                               OrgDb = "org.Hs.eg.db"),
                         just = c("left", "top"),
                         width = 3.5,
                         fontcolor = "black",
                         fill = "black",
                         range = c(0, scale.SNCA.max))


# Plot Low-C data SNCA in mDAN D30 (10kb resolution)
chr4.SNCA.sq = plotHicTriangle(
    data = mDAN.D30.25kb,
    params = region.p.chr4.small,
    height = 1.75,
    resolution = 25000,
    x = 4.5,
    y = 5.5,
    zrange = c(0, 5),
    just = c("left", "top"), 
    default.units = "inches",
    palette = colorRampPalette(brewer.pal(n = 9, "YlGnBu")))


# ATACseq signal smNPC_I and scale
ATAC_smNPC_I = plotSignal(
  data = bw.SNCA.smNPC.1, 
  params = region.p.chr4.small,
  fill = "#313695", 
  alpha = 0.7, 
  linecolor = NA,
  x = 4.5, 
  y = 7.25, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal smNPC_II and scale
ATAC_smNPC_II = plotSignal(
  data = bw.SNCA.smNPC.2, 
  params = region.p.chr4.small,
  fill = "#313695", 
  alpha = 0.6, 
  linecolor = NA,
  x = 4.5, 
  y = 7.25, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal smNPC_III and scale
ATAC_smNPC_III = plotSignal(
  data = bw.SNCA.smNPC.3, 
  params = region.p.chr4.small,
  fill = "#313695", 
  alpha = 0.5, 
  linecolor = NA,
  x = 4.5, 
  y = 7.25, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = ATAC_smNPC_III, 
          at = c(0, scale.SNCA.max),
          fontsize = 6)

plotText(label = "smNPC", 
         fontsize = 6, 
         fontcolor = "#313695",
         fontfamily = "Helvetica",
         x = 4.125, 
         y = 7.3125, 
         just = c("left", "top"),
         default.units = "inches",
         fontface = "bold")

# ATACseq signal mDAN D15 I and scale
ATAC_mDAN.D15_I = plotSignal(
  data = bw.SNCA.mDAN.D15.1, 
  params = region.p.chr4.small,
  fill = "#053061", 
  alpha = 0.7, 
  linecolor = NA,
  x = 4.5, 
  y = 7.625, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal mDAN D15 II and scale
ATAC_mDAN.D15_II = plotSignal(
  data = bw.SNCA.mDAN.D15.2, 
  params = region.p.chr4.small,
  fill = "#053061", 
  alpha = 0.6, 
  linecolor = NA,
  x = 4.5, 
  y = 7.625, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal mDAN D15 III and scale
ATAC_mDAN.D15_III = plotSignal(
  data = bw.SNCA.mDAN.D15.3, 
  params = region.p.chr4.small,
  fill = "#053061", 
  alpha = 0.5, 
  linecolor = NA,
  x = 4.5, 
  y = 7.625, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = ATAC_mDAN.D15_III, 
          at = c(0, scale.SNCA.max),
          fontsize = 6)

plotText(label = "mDAN D15", 
         fontsize = 6, 
         fontcolor = "#053061",
         fontfamily = "Helvetica",
         x = 4.125, 
         y = 7.6875, 
         just = c("left", "top"),
         default.units = "inches",
         fontface = "bold")

# ATACseq signal mDAN D30 I and scale
ATAC_mDAN.D30_I = plotSignal(
  data = bw.SNCA.mDAN.D30.1, 
  params = region.p.chr4.small,
  fill = "#2D004B", 
  alpha = 0.7, 
  linecolor = NA,
  x = 4.5, 
  y = 8.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal mDAN D30 II and scale
ATAC_mDAN.D30_II = plotSignal(
  data = bw.SNCA.mDAN.D30.2, 
  params = region.p.chr4.small,
  fill = "#2D004B", 
  alpha = 0.6, 
  linecolor = NA,
  x = 4.5, 
  y = 8.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal mDAN D30 III and scale
ATAC_mDAN.D30_III = plotSignal(
  data = bw.SNCA.mDAN.D30.3, 
  params = region.p.chr4.small,
  fill = "#2D004B", 
  alpha = 0.5, 
  linecolor = NA,
  x = 4.5, 
  y = 8.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = ATAC_mDAN.D30_III, 
          at = c(0, scale.SNCA.max),
          fontsize = 6)

plotText(label = "mDAN D30", 
         fontsize = 6, 
         fontcolor = "#2D004B",
         fontfamily = "Helvetica",
         x = 4.125, 
         y = 8.125, 
         just = c("left", "top"),
         default.units = "inches",
         fontface = "bold")

# ATACseq signal mDAN D50 I and scale
ATAC_mDAN.D50_I = plotSignal(
  data = bw.SNCA.mDAN.D50.1, 
  params = region.p.chr4.small,
  fill = "#003C30", 
  alpha = 0.7, 
  linecolor = NA,
  x = 4.5, 
  y =87.375, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal mDAN D50 II and scale
ATAC_mDAN.D50_II = plotSignal(
  data = bw.SNCA.mDAN.D50.2, 
  params = region.p.chr4.small,
  fill = "#003C30", 
  alpha = 0.6, 
  linecolor = NA,
  x = 4.5, 
  y = 8.375, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal mDAN D50 III and scale
ATAC_mDAN.D50_III = plotSignal(
  data = bw.SNCA.mDAN.D50.3, 
  params = region.p.chr4.small,
  fill = "#003C30", 
  alpha = 0.5, 
  linecolor = NA,
  x = 4.5, 
  y = 8.375, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = ATAC_mDAN.D50_III, 
          at = c(0, scale.SNCA.max),
          fontsize = 6)

plotText(label = "mDAN D50", 
         fontsize = 6, 
         fontcolor = "#003C30",
         fontfamily = "Helvetica",
         x = 4.125, 
         y = 8.5, 
         just = c("left", "top"),
         default.units = "inches",
         fontface = "bold")

# Add gene name and genome labels for the chr4
# Add gene name
SNCA_g = plotGenes(params = region.p.chr4.small,
          x = 4.5,
          y = 8.8125, 
          height = 0.5,
          geneHighlights = data.frame(gene = c("SNCA"),
                        color = "red"),
          assembly = assembly(Genome = "hg38refGene", 
                      TxDb = "TxDb.Hsapiens.UCSC.hg38.refGene", 
                      OrgDb = "org.Hs.eg.db"),
          geneBackground = "black")


# Add genome label
annoGenomeLabel(plot = SNCA_g, 
                scale = "Mb", 
                axis = "x",
                x = 4.5, 
                y = 8.6875, 
                just = c("left", "top"))

pageGuideHide()
dev.off()
```

Process the results of the SNEEP pipeline
```{r, eval = FALSE, echo = TRUE}
## Take the smNPCs, TH+ neurons at day 15 - day 30 - day 50
## Load the data
# smNPC
SNEEP.smNPC = read_delim("/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/parkinsonDisease_project_Deborah_Lasse/sneep_witATAC_mergedlowC_2409/result_PD_smNPV_lowC_ATAC_merged_2409.txt",
                         delim = "\t",
                         col_names = TRUE)

# Add a column specifying the sample name
SNEEP.smNPC = SNEEP.smNPC %>% 
  mutate(Sample = "smNPC")

# Unique rSNPs 
SNEEP.smNPC %>% 
  distinct(SNP_position) %>% 
  write_delim("/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/uniq_rSNPs_smNPC.txt",
              delim = "\t",
              col_names = FALSE)

# How many unique target genes associated to those 91 PD-rSNPs found in smNPC
TG_smNPC = SNEEP.smNPC %>% 
  dplyr::select(SNP_position, geneNames) %>% 
  dplyr::filter(geneNames != ".") %>% 
  group_by(SNP_position, geneNames) %>% 
  distinct(SNP_position, .keep_all = TRUE) %>% 
  separate_rows(geneNames, sep = ",") %>% 
  ungroup() %>% 
  distinct(geneNames)

write_delim(TG_smNPC,
            "/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/enrichR/distinct_target_genes_smNPC.txt",
              delim = "\t",
              col_names = FALSE)

# 293 target genes are associated to 91 PD-rSNPs in smNPC

# TH+ neurons day 15
SNEEP.THposD15 = read_delim("/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/parkinsonDisease_project_Deborah_Lasse/sneep_witATAC_mergedlowC_2409/result_PD_mDAN_day15_lowC_ATAC_N1_2409.txt",
                         delim = "\t",
                         col_names = TRUE)

SNEEP.THposD15 = SNEEP.THposD15 %>% 
  mutate(Sample = "THpos_neurons_D15")

# Unique rSNPs 
SNEEP.THposD15 %>% 
  distinct(SNP_position) %>% 
  write_delim("/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/uniq_rSNPs_mDAND15.txt",
              delim = "\t",
              col_names = FALSE)

# How many unique target genes associated to those 180 PD-rSNPs found in mDAN D15
TG_mDAN.D15 = SNEEP.THposD15 %>% 
  dplyr::select(SNP_position, geneNames) %>% 
  dplyr::filter(geneNames != ".") %>% 
  group_by(SNP_position, geneNames) %>% 
  distinct(SNP_position, .keep_all = TRUE) %>% 
  separate_rows(geneNames, sep = ",") %>% 
  ungroup() %>% 
  distinct(geneNames)

write_delim(TG_mDAN.D15,
            "/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/enrichR/distinct_target_genes_mDAN_D15.txt",
              delim = "\t",
              col_names = FALSE)

# 308 target genes are associated to 180 PD-rSNPs in mDAN D15

# TH+ neurons day 30
SNEEP.THposD30 = read_delim("/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/parkinsonDisease_project_Deborah_Lasse/sneep_witATAC_mergedlowC_2409/result_PD_mDAN_day30_lowC_ATAC_N1_2409.txt",
                         delim = "\t",
                         col_names = TRUE)

SNEEP.THposD30 = SNEEP.THposD30 %>% 
  mutate(Sample = "THpos_neurons_D30")

# Unique rSNPs 
SNEEP.THposD30 %>% 
  distinct(SNP_position) %>% 
  write_delim("/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/uniq_rSNPs_mDAND30.txt",
              delim = "\t",
              col_names = FALSE)

# How many unique target genes associated to those 181 PD-rSNPs found in mDAN D30
TG_mDAN.D30 = SNEEP.THposD30 %>% 
  dplyr::select(SNP_position, geneNames) %>% 
  dplyr::filter(geneNames != ".") %>% 
  group_by(SNP_position, geneNames) %>% 
  distinct(SNP_position, .keep_all = TRUE) %>% 
  separate_rows(geneNames, sep = ",") %>% 
  ungroup() %>% 
  distinct(geneNames)

write_delim(TG_mDAN.D30,
            "/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/enrichR/distinct_target_genes_mDAN_D30.txt",
              delim = "\t",
              col_names = FALSE)

# 301 target genes are associated to 181 PD-rSNPs in mDAN D30

# TH+ neurons day 50
SNEEP.THposD50 = read_delim("/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/parkinsonDisease_project_Deborah_Lasse/sneep_witATAC_mergedlowC_2409/result_PD_mDAN_day50_lowC_ATAC_N1_2409.txt",
                         delim = "\t",
                         col_names = TRUE)


SNEEP.THposD50 = SNEEP.THposD50 %>% 
  mutate(Sample = "THpos_neurons_D50")

# Unique rSNPs 
SNEEP.THposD50 %>% 
  distinct(SNP_position) %>% 
  write_delim("/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/uniq_rSNPs_mDAND50.txt",
              delim = "\t",
              col_names = FALSE)

# How many unique target genes associated to those 160 PD-rSNPs found in mDAN D50
TG_mDAN.D50 = SNEEP.THposD50 %>% 
  dplyr::select(SNP_position, geneNames) %>% 
  dplyr::filter(geneNames != ".") %>% 
  group_by(SNP_position, geneNames) %>% 
  distinct(SNP_position, .keep_all = TRUE) %>% 
  separate_rows(geneNames, sep = ",") %>% 
  ungroup() %>% 
  distinct(geneNames)

write_delim(TG_mDAN.D50,
            "/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/enrichR/distinct_target_genes_mDAN_D50.txt",
              delim = "\t",
              col_names = FALSE)

# 300 target genes are associated to 160 PD-rSNPs in mDAN D50

# Combine target genes from mDAN D15, D30 and D50 together
TG_distinct_mDAN = TG_mDAN.D15 %>% 
  bind_rows(TG_mDAN.D30) %>% 
  bind_rows(TG_mDAN.D50) %>% 
  distinct(geneNames)

write_delim(TG_distinct_mDAN,
            "/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/enrichR/distinct_target_genes_mDAN_all_together.txt",
              delim = "\t",
              col_names = FALSE)

# Combine them all
SNEEP_all = SNEEP.smNPC %>% 
  bind_rows(SNEEP.THposD15) %>% 
  bind_rows(SNEEP.THposD30) %>% 
  bind_rows(SNEEP.THposD50)

# And save them
write_delim(SNEEP_all,
            "/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/SNEEP_all_res_comb.txt",
            delim = "\t",
            col_names = TRUE)

# Load when needed
SNEEP_all = read_delim("/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/SNEEP_all_res_comb.txt",
            delim = "\t",
            col_names = TRUE)

SNEEP_all = read_delim("/Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/SNEEP_all_res_comb.txt",
            delim = "\t",
            col_names = TRUE)

# How many unique rSNPs per samples
SNEEP_all %>%
  group_by(Sample) %>%
  summarise(dplyr::n())

SNEEP_all %>%
  group_by(Sample) %>%
  distinct(SNP_position) %>% 
  summarise(dplyr::n())

# How many unique rSNPs?
SNEEP_all %>% 
  distinct(SNP_position) %>% 
  write_delim(.,
              "/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/SNEEP_uniq_rSNPs.txt",
              delim = "\t",
              col_names = TRUE)

# How many SNPs among the 254 PD-rSNPs are creating or disturbing TFBS
SNEEP_all %>% 
  distinct(SNP_position, .keep_all = TRUE) %>% 
  dplyr::select(SNP_position, log_pvalueBindAffVar1_pvalueBindAffVar2) %>% 
  mutate(pos = sum(log_pvalueBindAffVar1_pvalueBindAffVar2 > 0),
         neg = sum(log_pvalueBindAffVar1_pvalueBindAffVar2 < 0))

# How many SNPs among the 254 PD-rSNPs are associated to MAPT or SNCA
# MAPT
SNEEP_all %>% 
  #group_by(SNP_position, geneNames) %>% 
  dplyr::select(SNP_position, TF, geneNames) %>% 
  #distinct(SNP_position, .keep_all = TRUE) %>% 
  dplyr::filter(str_detect(geneNames, "MAPT")) %>% 
  distinct(SNP_position, .keep_all = TRUE) 

# 34 out 254 PD-rSNPs are associated to MAPT
# SNCA
SNEEP_all %>% 
  #group_by(SNP_position, geneNames) %>% 
  dplyr::select(SNP_position, TF, geneNames) %>% 
  #distinct(SNP_position, .keep_all = TRUE) %>% 
  dplyr::filter(str_detect(geneNames, "SNCA")) %>% 
  distinct(SNP_position, .keep_all = TRUE) 

# 7 out 254 PD-rSNPs are associated to SNCA

# How many chromosomes have at least one of the 254 PD-rSNPs 
SNEEP_all %>% 
  distinct(SNP_position) %>% 
  separate(SNP_position, into = c("chr", "tmp"), sep = ":") %>% 
  group_by(chr) %>% 
  summarise(chr_nb = n())
```

Now that unique set of rSNPs is available, filter this number by computing correlation between the expression of the TFs and their predicted target genes
```{r, eval = FALSE, echo = TRUE}
# Load RNAse data
dat.RPKM.filt = read_rds("/home/vagrant/Manuscript_1/FIGURE2/mat_RPKM.rds")
#dat.RPKM.filt = read_rds("/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE2/mat_RPKM.rds")

# Extract the TFBS predicted to be altered by the 254 rSNPs and plot their expression as well a and s the expression of their putative target genes
## As some TFs act as dimers, extract the expression of each and the lowest one (of each pair) will be the rate limiting TF
TF_dimer_exp = SNEEP_all %>% 
  dplyr::select(SNP_position:var2, TF, geneNames, Sample) %>% 
  group_by(SNP_position, Sample) %>% 
  dplyr::filter(geneNames != ".") %>% # Filter out rSNPs where there are no putative target genes
  mutate(TF = gsub("\\(.*", "\\1", TF)) %>% # remove the JASPAR ID next to some TFs
  dplyr::filter(str_detect(TF, "::")) %>%  # Filter out the TF dimers and check if one of the TFs is low expressed
  dplyr::select(SNP_position, TF) %>% 
  mutate(TF_dimer = TF) %>% 
  separate_rows(TF, sep = "::") %>% 
  ungroup() %>% 
  dplyr::select(-Sample)

## Plot
dat_2_plot = dat.RPKM.filt %>% 
  dplyr::filter(gene_name %in% TF_dimer_exp$TF) %>% 
  dplyr::rename(TF = gene_name) %>% 
  left_join(TF_dimer_exp,
            by = "TF") %>% 
  mutate(TF = as.factor(TF))

pdf("/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/TF_dimer_Ind_exp.pdf",
    width = 15,
    height = 20)
ggboxplot(dat_2_plot,
          x = "Cond", 
          y = "RPKM", 
          add = "jitter", 
          color = "TF", 
          #palette = c("#A9A9A9", "#B22222", "#DC143C", "#E9967A"),
          facet.by = "TF_dimer",
          xlab = "", 
          ylab = "Reads Per Kilobase Million (RPKM)") +
  geom_hline(yintercept = 5,
             linetype = "dashed") +  # 5 RPKM expression threshold
  scale_x_discrete(labels = c("smNPC",
                              expression("mDAN D15"),
                              expression("mDAN D30"),
                              expression("mDAN D50"))) +
  theme(legend.position = "right",
        axis.text.x = element_text(angle = 90))

dev.off()

## Compute Pearson correlation to filter rSNPs
# Tibble for TF expression
TF_exp = SNEEP_all %>%
  dplyr::select(SNP_position, TF, geneNames) %>%
  dplyr::filter(geneNames != ".") %>% # no target genes idenditifed
  mutate(TF = gsub("\\(.*", "\\1", TF)) %>%
  separate_rows(TF, sep = "::") %>% # Separate TF dimers into single TF
  separate_rows(geneNames, sep = ",") %>% # Separate target genes into single target genes
  unite(col = "Item", c("SNP_position", "TF", "geneNames"), remove = FALSE) %>%
  dplyr::select(Item, TF) %>%
  dplyr::rename(gene_name = TF) %>%
  dplyr::filter(gene_name %in% dat.RPKM.filt$gene_name) %>%
  left_join(dat.RPKM.filt,
            by = "gene_name") %>%
  group_by(gene_name) %>% 
  dplyr::filter(any(RPKM > 1)) %>% # Keep only TF that have RPKM > 1 in all samples
  dplyr::select(-gene_id, -Cond) %>%
  dplyr::rename(TF = gene_name,
                RPKM.TF = RPKM)

# Tibble for TG expression  
TG_exp = SNEEP_all %>%
  dplyr::select(SNP_position, TF, geneNames) %>%
  dplyr::filter(geneNames != ".") %>% # no target genes idenditifed
  mutate(TF = gsub("\\(.*", "\\1", TF)) %>%
  separate_rows(TF, sep = "::") %>% # Separate TF dimers into single TF
  separate_rows(geneNames, sep = ",") %>% # Separate target genes into single target genes
  unite(col = "Item", c("SNP_position", "TF", "geneNames"), remove = FALSE) %>%
  dplyr::select(Item, geneNames) %>%
  separate_rows(geneNames, sep = ",") %>%
  dplyr::rename(gene_name = geneNames) %>%
  dplyr::filter(gene_name %in% dat.RPKM.filt$gene_name) %>%
  left_join(dat.RPKM.filt,
            by = "gene_name") %>%
  group_by(gene_name) %>% 
  dplyr::filter(any(RPKM > 1)) %>% # Keep only target genes that have RPKM > 1 in all samples
  dplyr::select(-gene_id, -Cond) %>%
  dplyr::rename(target_gene = gene_name,
                RPKM.TG = RPKM)

# Join both target gene and TF tables. Filter out TF or target genes with expression below 1 RPKM
TF_TG_mat = merge(TF_exp, TG_exp) %>% 
  as_tibble() %>% 
  group_by(TF, target_gene) %>% 
  distinct(Sample, .keep_all = TRUE)

# Pearson correlation
TF_TG_mat_Pearson_cor = cor_test(TF_TG_mat %>% 
                                   group_by(TF, target_gene),
                                 vars = c("RPKM.TF","RPKM.TG"),
                                 use = "everything")

write_delim(TF_TG_mat_Pearson_cor,
            "/Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/TF_TG_exp_pearsonCor.txt",
            delim = "\t",
            col_names = TRUE)

# Filter the correlation matrix to only keep TF-target gene pairs with a strong correlation/anti-correlation
TF_TG_mat_Pearson_cor.filt = TF_TG_mat_Pearson_cor %>% 
  dplyr::filter(cor > 0.5 | cor < -0.5)

unique(TF_TG_mat_Pearson_cor.filt$target_gene)
unique(TF_TG_mat_Pearson_cor.filt$TF)

write_delim(TF_TG_mat_Pearson_cor.filt,
            "/Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/TF_TG_exp_pearsonCor_abs0.5.txt",
            delim = "\t",
            col_names = TRUE)

# Reassociate the rSNPs to which the TF-target genes pairs belongs and make a table
tbl.TF.TG = TF_TG_mat_Pearson_cor.filt %>% 
  unite(col = "TF_TG_pair", TF, target_gene, sep = "_", remove = FALSE)

tbl.TF.TG.FINAL = TF_TG_mat %>% 
  ungroup() %>% 
  dplyr::filter(str_detect(Item, paste0(tbl.TF.TG$TF_TG_pair, collapse = "|"))) %>% 
  distinct(Item, .keep_all = TRUE) %>% 
  dplyr::select(Item, TF, target_gene) %>% 
  mutate(rSNPs = gsub("\\_.*", "\\1", Item)) %>% 
  dplyr::select(rSNPs, TF, target_gene)

# Select the coordinates of the rSNPs ans associate back rsID from UCSC using all snps from db155
tbl.TF.TG.FINAL %>% 
  dplyr::select(rSNPs) %>% 
  distinct(rSNPs) %>% 
  write_delim(.,
              "/Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/UCSC_coordinates_rSNPs.txt",
              delim = "\t",
              col_names = FALSE)

# Load the newly created file
# rsID_PD_rSNP = read_delim("/Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/UCSC_coordinates_rSNPs_rsID.txt",
#               delim = ",",
#               col_names = TRUE)

rsID_PD_rSNP = read_delim("/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/UCSC_coordinates_rSNPs_rsID.txt",
              delim = ",",
              col_names = TRUE)

# Load the newly created file
rsID_PD_rSNP.filt = rsID_PD_rSNP %>% 
  dplyr::filter(class == "snv") %>% 
  unite(col = "tmp", c(`#"chrom"`, chromStart), sep = ":", remove = TRUE) %>% 
  unite(col = "rSNPs", c("tmp", "chromEnd"), sep = "-", remove = FALSE) %>% 
  dplyr::select(rSNPs, name, ref, alts) %>% 
  distinct(name, .keep_all = TRUE) # Remove doublets

# Filter the final table
tbl.TF.TG.FINAL.rsID = tbl.TF.TG.FINAL %>% 
  #dplyr::select(rSNPs) %>% 
  #distinct(rSNPs) %>% 
  dplyr::filter(rSNPs %in% rsID_PD_rSNP.filt$rSNPs) %>% 
  left_join(rsID_PD_rSNP.filt,
            by = "rSNPs") %>% 
  dplyr::select(rSNPs, name:alts, TF, target_gene) %>% 
  separate(rSNPs, into = c("chr", "tmp"), sep = ":", remove = TRUE) %>% 
  separate(tmp, into = c("start", "end"), sep = "-", remove = TRUE) %>% 
  dplyr::rename(rsID = name)

# Extract the unique rsIDs and get the MAF (minor allele frequencies)
snp2MAF = tbl.TF.TG.FINAL.rsID %>% 
  dplyr::select(rsID) %>% 
  distinct(rsID) %>% 
  pull()

# Get allele frequencies
LDproxy_batch(snp = snp2MAF,
              pop = "EUR",
              r2d = "r2",
              token = "5bba40af90d7",
              append = TRUE,
              genome_build = "grch38_high_coverage")

# The data has been automaticcaly saved -> load it
MAF.snp = read_delim("/home/vagrant/Manuscript_1/TABLES/combined_query_snp_list_grch38_high_coverage.txt",
                     delim = "\t",
                     col_names = TRUE)

# MAF.snp = read_delim("/Volumes/deborah.gerard/Documents/Manuscript_1/TABLES/combined_query_snp_list_grch38_high_coverage.txt",
#                      delim = "\t",
#                      col_names = TRUE)

# The linkage desiquilibrium linkage value has been also calculated. Filter only the snp that are in linkage desiquilibrium with themselves
MAF.snp %>% 
  dplyr::select(RS_Number:Distance) %>% 
  dplyr::rename(query_snp = RS_Number,
                rsID = Coord,
                Coord = Alleles,
                Allele = MAF,
                MAF = Distance) %>% 
  mutate(same.snp = if_else(query_snp == rsID, 1, 0)) %>% 
  # dplyr::filter(same.snp == 1,
  #               rsID %in% snp2MAF) %>% 
  dplyr::filter(same.snp == 1) %>% 
  #left_join(tbl.TF.TG.FINAL.rsID, by = "rsID") %>% 
  full_join(tbl.TF.TG.FINAL.rsID, by = "rsID") %>% 
  #distinct(rsID, .keep_all = TRUE) %>% 
  View()
  write_delim(.,
            "/home/vagrant/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/final_tbl_PD_rSNPs_with_MAF.txt",
            delim = "\t",
            col_names = TRUE)

# Expression distribution of the 77 target genes
#unique(TF_TG_mat_Pearson_cor.filt$target_gene)
pdf("/Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/ExpDistri77TG.pdf",
    width = 10,
    height = 13)

TF_TG_mat %>% 
  ungroup() %>% 
  dplyr::select(Sample, target_gene, RPKM.TG) %>% 
  group_by(Sample) %>% 
  distinct(target_gene, .keep_all = TRUE) %>% 
  dplyr::filter(target_gene %in% TF_TG_mat_Pearson_cor.filt$target_gene) %>% 
  mutate(target_gene = as.factor(target_gene),
         Cond = gsub("\\_.*", "\\1", Sample),
         log2RPKM.TG = log2(RPKM.TG + 0.001)) %>% # add a pseudocount to avoid -Inf if RPKM is 0
  ggdensity(.,
            x = "log2RPKM.TG",
            facet.by = "target_gene",
            fill = "target_gene",
            title = "Distribution of the log2 expression of the 77 target genes") +
  theme(legend.position = "none") +
  geom_vline(xintercept = 0,
             linetype = "dashed")

dev.off()

# Plot of the Pearson correlation coefficient vs the pvalue of the diffbinf affinity
# Take the necessary columns from SNEEP_all
SNEEP_all_2 = SNEEP_all %>% 
  mutate(TF = gsub("\\(.*", "\\1", TF)) %>% 
  dplyr::select(SNP_position, TF, geneNames, pvalue_DiffBindAff) %>% 
  dplyr::filter(geneNames != ".") %>% 
  separate_rows(., geneNames, sep = ",") %>% 
  unite(col = "Item", SNP_position, TF, geneNames, sep = "_", remove = FALSE) 

# And merge with the table with the Pearson correlation
TF_TG_mat_Pearson_cor2 = TF_TG_mat_Pearson_cor %>% 
  unite(col = "TF_TG_pair", TF, target_gene, sep = "_", remove = TRUE)

lbl.pl = c("LHX1_BAG3",
           "ZBTB14_IDUA",
           "NR2C2_SCARB2")

pvalDiffBind_cor = TF_TG_mat %>% 
  ungroup() %>% 
  dplyr::select(Item) %>% 
  mutate(SNP_position = gsub("\\_.*", "\\1", Item)) %>% 
  distinct(Item, .keep_all = TRUE) %>% 
  left_join(SNEEP_all_2,
            by = "Item") %>% 
  unite(col = "TF_TG_pair", TF, geneNames, sep = "_", remove = TRUE) %>%
  left_join(TF_TG_mat_Pearson_cor2, 
            by = "TF_TG_pair") %>% 
  #drop_na() %>% 
  dplyr::select(Item, TF_TG_pair, pvalue_DiffBindAff, cor) %>% 
  distinct(Item, .keep_all = TRUE) %>% 
  mutate(pvalue_DiffBindAff_log = log2(pvalue_DiffBindAff),
         lbl = if_else(TF_TG_pair %in% lbl.pl, TF_TG_pair, NA_character_))

# And plot
ggscatter(pvalDiffBind_cor,
          x = "cor",
          y = "pvalue_DiffBindAff_log",
          color = "lbl", 
          #palette = c("#00AFBB", "black"),
          label = "lbl",
          repel = TRUE)
  
```

```{r, eval = FALSE, echo = TRUE}

###################
#### Figure 1D ####
###################

# Locus plot
# Here our SNPs of interest and their location
snp = GRanges(seqnames = c("chr10", "chr4", "chr4"),
              ranges = IRanges(start = c(119651404, 987107, 76213716),
                               end = c(119651405, 987108, 76213717)))

# Retrieve them from Nalls et al data
snp_Nalls = subsetByOverlaps(nalls_allSNPs.HG38.GR, snp) %>% 
  as_tibble() %>% 
  dplyr::select(-width, -strand) %>% 
  mutate(rsID = c("rs144814361", "rs1465922", "rs11248061"))

# Visualise the genomic region per SNP (+- 400100 bp)
# rs144814361
r = 40100

# Filter for the SNP in BAG3 regulatory region
Nalls_sel_reg.BAG3 = nalls_allSNPs.HG38.GR %>% 
  as_tibble() %>% 
  dplyr::filter(seqnames == "chr10",
         between(base_pair_position, 119651405 - r, 119651405 + r)) %>% 
  mutate(minus_log_10_p = -log10(p),
         GWAS_p_sig = if_else(minus_log_10_p > -log10(5e-8), "yes", "no"),
         rsID = if_else(seqnames == "chr10" & base_pair_position == 119651405, "rs144814361", ""))
  
# Get SNPs info from Ensembl
#snp.ensembl = useEnsembl(biomart = "snp", 
#                         dataset = "hsapiens_snp")

snp.ensembl = useMart(biomart = "ENSEMBL_MART_SNP",
                      dataset = "hsapiens_snp")

# SNP info
query.snp = getBM(c("ensembl_gene_stable_id", 
                            "refsnp_id", 
                            "chr_name", 
                            "chrom_start", 
                            "chrom_end", 
                            "minor_allele", 
                            "minor_allele_freq"),
                          filters = "snp_filter",
                          values = c("rs144814361", "rs1465922", "rs11248061"),
                  mart = snp.ensembl)

# Get rsID of SNPs that are significant and append 
position_chr10 = Nalls_sel_reg.BAG3 %>% 
  mutate(chr = gsub("chr", "", seqnames)) %>% 
  unite(col = "position", c(chr, end, base_pair_position), sep = ":", remove = FALSE) %>% 
  dplyr::filter(GWAS_p_sig != "no") %>%
  dplyr::select(position) %>% 
  pull()


rsID_sig_SNPs.chr10 = getBM(c("refsnp_id", "chr_name", "chrom_start", "chrom_end", "allele", "minor_allele"),
                      filters = "chromosomal_region",
                      values = position_chr10,
                      mart = snp.ensembl)

rsID_sig_SNPs.chr10 = rsID_sig_SNPs.chr10 %>% 
  unite(col = "position", chr_name:chrom_end, sep = ":", remove = FALSE)

Nalls_sel_reg.BAG3 = Nalls_sel_reg.BAG3 %>% 
  mutate(chr = gsub("chr", "", seqnames)) %>% 
  unite(col = "position", c(chr, end, base_pair_position),
        sep = ":", remove = FALSE) %>% 
  left_join(rsID_sig_SNPs.chr10, by = "position")

# Calculate R2 to check if SNPs are in LD
BAG3_SNPs_LD = Nalls_sel_reg.BAG3 %>% 
  dplyr::select(refsnp_id) %>% 
  drop_na()

BAG3_R2 = LDproxy(snp = "rs144814361",
                   pop = "EUR",
                   r2d = "r2",
                   token = "5bba40af90d7",
                   genome_build = "grch38_high_coverage")

BAG3_R2 = BAG3_R2 %>% 
  as_tibble() %>% 
  dplyr::select(RS_Number, Coord, R2) %>% 
  dplyr::rename(refsnp_id = RS_Number)

# Plot
BAG3.pl = Nalls_sel_reg.BAG3 %>% 
  unite("Coord", seqnames, base_pair_position, sep = ":", remove = FALSE) %>% 
  left_join(BAG3_R2, by = "Coord") %>% 
  dplyr::select(seqnames, base_pair_position, p, refsnp_id.x, R2) %>% 
  dplyr::rename(chrom = seqnames,
               pos = base_pair_position,
               snp = refsnp_id.x, 
               LD = R2) %>% 
  mutate(chrom = as.character(chrom)) %>% 
  group_by(LD) %>% 
  mutate(LD_grp = cut(LD, seq(0, 1.0, 0.2)))

BAG3.pl$LD_grp = addNA(BAG3.pl$LD_grp)

BAG3_leadSNP = "rs144814361"

BAG3_MP = plotManhattan(data = BAG3.pl,
              chrom = "chr10",
              chromstart = 119611405,
              chromend = 119691405,
              assembly = "hg38",
              fill = colorby("LD_grp",
                             palette = colorRampPalette(c("#35b779",
                                                          "#21918c", 
                                                          "#31688e",
                                                          "#443983", 
                                                          "#440154", 
                                                          "#CCCCCC"))),
        trans = "-log10", 
        sigLine = TRUE, col = "grey",
        lty = 2, range = c(0, 16),
        leadSNP = list(
        snp = BAG3_leadSNP,
        pch = 18,
        cex = 0.75,
        fill = "#990066",
        fontsize = 8),
              x = 0.5, 
              y = 8.5, 
              width = 1.5, 
              height = 2.5,
              just = c("left", "top"), 
              default.units = "inches")

## Annotate genome label
annoGenomeLabel(
    plot = BAG3_MP, x = 0.5, y = 11,
    fontsize = 8, scale = "Mb",
    just = c("left", "top"), default.units = "inches"
)

## Annotate y-axis
annoYaxis(
    plot = BAG3_MP,
    at = c(0, 2, 4, 6, 8, 10, 12),
    axisLine = TRUE, fontsize = 8
)

## Plot y-axis label
plotText(
    label = "-log10(p-value)", x = 0.125, y = 10.125, rot = 90,
    fontsize = 8, fontface = "bold", just = "center",
    default.units = "inches"
)

# Plot gene BAG3
plotGenes(
    chrom = "chr10", chromstart = 119611405, chromend = 119691405,
    x = 0.5, y = 8.5, width = 1.5, height = 0.5,
    just = c("left", "top"), default.units = "inches",
    fontcolor = "black",
    fill = "black",
    assembly = assembly(Genome = "hg38refGene", 
                      TxDb = "TxDb.Hsapiens.UCSC.hg38.refGene", 
                      OrgDb = "org.Hs.eg.db")
)

# Filter for the SNP in IDUA regulatory region
Nalls_sel_reg.IDUA = nalls_allSNPs.HG38.GR %>% 
  as_tibble() %>% 
  dplyr::filter(seqnames == "chr4",
         between(base_pair_position, 987108 - r, 987108 + r)) %>% 
  mutate(minus_log_10_p = -log10(p),
         GWAS_p_sig = if_else(minus_log_10_p > -log10(5e-8), "yes", "no"),
         rsID = if_else(seqnames == "chr4" & base_pair_position == 987108, "rs11248061", ""))

# Get rsID of SNPs that significant and append 
position_chr4.IDUA = Nalls_sel_reg.IDUA %>% 
  mutate(chr = gsub("chr", "", seqnames)) %>% 
  unite(col = "position", c(chr, end, base_pair_position), sep = ":", remove = FALSE) %>% 
  dplyr::filter(GWAS_p_sig != "no") %>%
  dplyr::select(position) %>% 
  pull()

rsID_sig_SNPs.chr4.IDUA = getBM(c("refsnp_id", "chr_name", "chrom_start", "chrom_end", "allele", "minor_allele",
                        "minor_allele_freq"),
                      filters = "chromosomal_region",
                      values = position_chr4.IDUA,
                      mart = snp.ensembl)

rsID_sig_SNPs.chr4.IDUA = rsID_sig_SNPs.chr4.IDUA %>% 
  unite(col = "position", chr_name:chrom_end, sep = ":", remove = FALSE)

Nalls_sel_reg.IDUA = Nalls_sel_reg.IDUA %>% 
  mutate(chr = gsub("chr", "", seqnames)) %>% 
  unite(col = "position", c(chr, end, base_pair_position),
        sep = ":", remove = FALSE) %>% 
  left_join(rsID_sig_SNPs.chr4.IDUA, by = "position")

# Calculate R2 to check if SNPs are in LD
IDUA_SNPs_LD = Nalls_sel_reg.IDUA %>% 
  dplyr::select(refsnp_id) %>% 
  drop_na()

IDUA_SNPs_LD = IDUA_SNPs_LD %>% 
  dplyr::filter(!str_detect(refsnp_id, "^CR"))

IDUA_R2 = LDproxy(snp = "rs11248061",
                   pop = "EUR",
                   r2d = "r2",
                   token = "5bba40af90d7",
                   genome_build = "grch38_high_coverage")

IDUA_R2 = IDUA_R2 %>% 
  as_tibble() %>% 
  dplyr::select(RS_Number, Coord, R2) %>% 
  dplyr::rename(refsnp_id = RS_Number)


# Plot
IDUA.pl = Nalls_sel_reg.IDUA %>% 
  unite("Coord", seqnames, base_pair_position, sep = ":", remove = FALSE) %>% 
  distinct(Coord, .keep_all = TRUE) %>% 
  left_join(IDUA_R2, by = "Coord") %>% 
  dplyr::select(seqnames, base_pair_position, p, refsnp_id.x, R2) %>% 
  dplyr::rename(chrom = seqnames,
               pos = base_pair_position,
               snp = refsnp_id.x, 
               LD = R2) %>% 
  mutate(chrom = as.character(chrom)) %>% 
  group_by(LD) %>% 
  mutate(LD_grp = cut(LD, seq(0, 1.0, 0.2)))

IDUA.pl$LD_grp = addNA(IDUA.pl$LD_grp)

IDUA_leadSNP = "rs11248061"

IDUA_MP = plotManhattan(data = IDUA.pl,
              chrom = "chr4",
              chromstart = 947108,
              chromend = 1027108,
              assembly = "hg38",
              fill = colorby("LD_grp",
                             palette = colorRampPalette(c("#35b779",
                                                          "#21918c", 
                                                          "#31688e",
                                                          "#443983", 
                                                          "#440154", 
                                                          "#CCCCCC"))),
        trans = "-log10", 
        sigLine = TRUE, col = "grey",
        lty = 2, range = c(0, 16),
        leadSNP = list(
        snp = IDUA_leadSNP,
        pch = 18,
        cex = 0.75,
        fill = "#990066",
        fontsize = 8),
              x = 2.5, 
              y = 8.5, 
              width = 1.5, 
              height = 2.5,
              just = c("left", "top"), 
              default.units = "inches")

## Annotate genome label
annoGenomeLabel(
    plot = IDUA_MP, x = 2.5, y = 11,
    fontsize = 8, scale = "Mb",
    just = c("left", "top"), default.units = "inches"
)

## Annotate y-axis
annoYaxis(
    plot = IDUA_MP,
    at = c(0, 2, 4, 6, 8, 10, 12),
    axisLine = TRUE, fontsize = 8
)

## Plot y-axis label
plotText(
    label = "-log10(p-value)", x = 2.125, y = 10.125, rot = 90,
    fontsize = 8, fontface = "bold", just = "center",
    default.units = "inches"
)

# Plot gene IDUA
plotGenes(
    chrom = "chr4", chromstart = 947108, chromend = 1027108,
    x = 2.5, y = 8.5, width = 1.5, height = 0.5,
    just = c("left", "top"), default.units = "inches",
    fontcolor = "black",
    fill = "black",
    assembly = assembly(Genome = "hg38refGene", 
                      TxDb = "TxDb.Hsapiens.UCSC.hg38.refGene", 
                      OrgDb = "org.Hs.eg.db")
)

# Filter for the SNP in SCARB2 regulatory region
Nalls_sel_reg.SCARB2 = nalls_allSNPs.HG38.GR %>% 
  as_tibble() %>% 
  dplyr::filter(seqnames == "chr4",
         between(base_pair_position, 76213717 - r, 76213717 + r)) %>% 
  mutate(minus_log_10_p = -log10(p),
         GWAS_p_sig = if_else(minus_log_10_p > -log10(5e-8), "yes", "no"),
         rsID = if_else(seqnames == "chr4" & base_pair_position == 76213717, "rs1465922", ""))

# Get rsID of SNPs that significant and append 
position_chr4.SCARB2 = Nalls_sel_reg.SCARB2 %>% 
  mutate(chr = gsub("chr", "", seqnames)) %>% 
  unite(col = "position", c(chr, end, base_pair_position), sep = ":", remove = FALSE) %>% 
  dplyr::filter(GWAS_p_sig != "no") %>%
  dplyr::select(position) %>% 
  pull()

rsID_sig_SNPs.chr4.SCARB2 = getBM(c("refsnp_id", "chr_name", "chrom_start", "chrom_end", "allele", "minor_allele",
                        "minor_allele_freq"),
                      filters = "chromosomal_region",
                      values = position_chr4.SCARB2,
                      mart = snp.ensembl)

rsID_sig_SNPs.chr4.SCARB2 = rsID_sig_SNPs.chr4.SCARB2 %>% 
  unite(col = "position", chr_name:chrom_end, sep = ":", remove = FALSE)

Nalls_sel_reg.SCARB2 = Nalls_sel_reg.SCARB2 %>% 
  mutate(chr = gsub("chr", "", seqnames)) %>% 
  unite(col = "position", c(chr, end, base_pair_position),
        sep = ":", remove = FALSE) %>% 
  left_join(rsID_sig_SNPs.chr4.SCARB2, by = "position")

# Calculate R2 to check if SNPs are in LD
SCARB2_SNPs_LD = Nalls_sel_reg.SCARB2 %>% 
  dplyr::select(refsnp_id) %>% 
  drop_na()

SCARB2_R2 = LDproxy(snp = "rs1465922",
                   pop = "EUR",
                   r2d = "r2",
                   token = "5bba40af90d7",
                   genome_build = "grch38_high_coverage")

SCARB2_R2 = SCARB2_R2 %>% 
  as_tibble() %>% 
  dplyr::select(RS_Number, Coord, R2) %>% 
  dplyr::rename(refsnp_id = RS_Number)

# Plot
SCARB2.pl = Nalls_sel_reg.SCARB2 %>% 
  unite("Coord", seqnames, base_pair_position, sep = ":", remove = FALSE) %>% 
  left_join(SCARB2_R2, by = "Coord") %>% 
  dplyr::select(seqnames, base_pair_position, p, refsnp_id.x, R2) %>% 
  dplyr::rename(chrom = seqnames,
               pos = base_pair_position,
               snp = refsnp_id.x, 
               LD = R2) %>% 
  mutate(chrom = as.character(chrom)) %>% 
  group_by(LD) %>% 
  mutate(LD_grp = cut(LD, seq(0, 1.0, 0.2)))

SCARB2.pl$LD_grp = addNA(SCARB2.pl$LD_grp)

SCARB2_leadSNP = "rs1465922"

SCARB2_MP = plotManhattan(data = SCARB2.pl,
              chrom = "chr4",
              chromstart = 76173717,
              chromend = 76253717,
              assembly = "hg38",
              fill = colorby("LD_grp",
                             palette = colorRampPalette(c("#35b779",
                                                          "#21918c", 
                                                          "#31688e",
                                                          "#443983", 
                                                          "#440154", 
                                                          "#CCCCCC"))),
        trans = "-log10", 
        sigLine = TRUE, col = "grey",
        lty = 2, range = c(0, 16),
        leadSNP = list(
        snp = SCARB2_leadSNP,
        pch = 18,
        cex = 0.75,
        fill = "#990066",
        fontsize = 8),
              x = 4.5, 
              y = 8.5, 
              width = 1.5, 
              height = 2.5,
              just = c("left", "top"), 
              default.units = "inches")

## Annotate genome label
annoGenomeLabel(
    plot = SCARB2_MP, x = 4.5, y = 11,
    fontsize = 8, scale = "Mb",
    just = c("left", "top"), default.units = "inches"
)

## Annotate y-axis
annoYaxis(
    plot = SCARB2_MP,
    at = c(0, 2, 4, 6, 8, 10, 12),
    axisLine = TRUE, fontsize = 8
)

## Plot y-axis label
plotText(
    label = "-log10(p-value)", x = 4.125, y = 10.125, rot = 90,
    fontsize = 8, fontface = "bold", just = "center",
    default.units = "inches"
)

# Plot gene SCARB2
plotGenes(
    chrom = "chr4", chromstart = 76173717, chromend = 76253717,
    x = 4.5, y = 8.5, width = 1.5, height = 0.5,
    just = c("left", "top"), default.units = "inches",
    fontcolor = "black",
    fill = "black",
    assembly = assembly(Genome = "hg38refGene", 
                      TxDb = "TxDb.Hsapiens.UCSC.hg38.refGene", 
                      OrgDb = "org.Hs.eg.db")
)

# Legend LD score
plotLegend(
  legend = c("LD reference SNP",
             paste("0", "<", "r^2", "<= 0.2"),
             paste("0.2", "<", "r^2", "<= 0.4"),
             paste("0.4", "<", "r^2", "<= 0.6"),
             paste("0.6", "<", "r^2", "<= 0.8"),
             paste("0.8", "<", "r^2", "<= 1.0"),
             "no LD data"),
    fill = c("#990066", "#35b779", "#21918c", "#31688e", "#443983", "#440154", "#CCCCCC"), 
  cex = 0.75,
    pch = c(18, 19, 19, 19, 19, 19, 19), border = FALSE, x = 6.5, y = 9.25,
    width = 1.0, height = 1.0, just = c("left", "top"),
    default.units = "inches"
)


dev.off()
```


### FIGURE 2
The SNEEP analysis will be run again by Nina with the PD GWAS SNPs that been filtered only for the GWAS p-value of 5e-08
```{r, eval = FALSE, echo = TRUE}
# This is the file obtained by Jochen
nalls_allSNPs = vroom(
  "/Volumes/deborah.gerard/Documents/epifunc/SNPs_overlap/nallsEtAl2019_hg38_full.tsv")

# Select the coordinates, the alleles and the p-value columns. Filter for the GWAS p-value
nalls_allSNPs %>% 
  dplyr::select(CHR_hg38:END_hg38, A1:A2, p) %>% 
  dplyr::filter(p < 5e-08) %>% # GWAS p-value threshold
  write_delim("/Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/input/Nalls_PD_SNPs_GWASpval_filtered.txt",
              delim = "\t",
              col_names = TRUE)

# Provide the RPKM data from the RNAseq to Nina
dat.RPKM = read_delim("/Volumes/deborah.gerard/Documents/epifunc/RNAseq/RPKM_genename",
                      delim = "\t", 
                      col_names = TRUE)

# Provide the file as a text file
dat.RPKM %>% 
  write_delim(.,
              "/Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/input/RNAseq/All_samples_RPKM.txt",
              delim = "\t",
              col_names = TRUE)

# Check the hg38 coordinates 
nalls_allSNPs %>% 
  dplyr::filter(CHR_hg38 == "chr4",
                END_hg38 == "76213717")
```

For the SNEEP analysis, retrieve also the `narrowPeak` files and send them to Nina
```{bash, eval = FALSE, echo = TRUE}
# narrowPeak files for smNPC, mDAN neurons day 15 - day 30 - day 50
rsync -avzPhu --no-p iris-cluster:/work/projects/epifunc/manuscript_sample_submission/ATAC/peaks/*narrowPeak /Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/input/ATAC/narrowPeaks/

# narrowPeak files for astrocytes and non-mDAN neurons at day 15 and day 50 (day 30 data does not exist)
# According to the epifunc gitlab (https://git-r3lab.uni.lu/jochen.ohnmacht/epifunc) they are in /work/projects/epifunc/genrich_replicates_opt. According the epifunc gitlab, the filtered.narrowPeak files were used for HINT-ATAC and EPIC-DREM
rsync -avzPhu --no-p iris-cluster:/work/projects/epifunc/genrich_replicates_opt/*neg.filtered.narrowPeak /Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/input/ATAC/narrowPeaks/

rsync -avzPhu --no-p iris-cluster:/work/projects/epifunc/genrich_replicates_opt/astrocytes.filtered.narrowPeak /Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/input/ATAC/narrowPeaks/
```

Create the page layout that will contain all the necessary plots

```{r fig2, eval = FALSE, echo = TRUE}
####################
#### Figure 2  ####
###################
# Save as TIFF, 300 ppi
tiff("/home/vagrant/Manuscript_1/FIGURE2/FIGURE2.tiff",
     width = 8.27,
     height = 11.67,
     units = "in",
     res = 300,
     compression = "lzw")

# Create a A4 blank page
pageCreate(width = 8.27, 
           height = 11.67, 
           default.units = "inches",
           showGuides = TRUE)

# text Figure 2
plotText(label = "Figure 2", 
         fontsize = 14,
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 0.25, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

#### PANEL A - SNEEP PIPELINE ####
# text A
plotText(label = "A", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 0.5, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# Load the picture prepared in Biorender
Fig2A_pip = readPNG("/home/vagrant/Manuscript_1/FIGURE2/Fig2A_FlowChart.png")

# Plot Figure 2A - Pipeline
plotRaster(image = Fig2A_pip,
           x = 0.5,
           y = 0.625,
           width = 3.5,
           height = 3.0,
           just = c("left", 
                    "top"),
           interpolate = TRUE)

#### PANEL B - Pizza plot of 50 PD-rSNPs (among 54) that shows allelic imbalance #### 
# text B
plotText(label = "B", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 4.125, 
         y = 0.5, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# Since there are 254 unique rSNPs but we do not have about them being heterozygous or not, check if there is also chromatin allelic imbalance
# hg38
# Load the 254 PD rSNPs coordinates
PD_rSNPs_254 = read_delim("/Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/SNEEP_uniq_rSNPs.txt",
                          delim = "\t",
                          col_names = TRUE)

PD_rSNPs_254_fmt = PD_rSNPs_254 %>% 
  separate(SNP_position, into = c("chr", "tmp"), sep = ":", remove = FALSE) %>% 
  separate(tmp, into = c("start", "end"), sep = "-", remove = TRUE) %>% 
  mutate(start = as.numeric(start),
         end = as.numeric(end))
  
# Make a Granges object of the 254 PD rSNPs (width has to be one)
searchArea = GRanges(seqnames = PD_rSNPs_254_fmt$chr, 
                     ranges = IRanges(start = PD_rSNPs_254_fmt$end, 
                                      end = PD_rSNPs_254_fmt$end),
                     name = PD_rSNPs_254_fmt$SNP_position)

# Path where the ATACseq BAM files are
pathToFiles = "/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE4/"

# Import a specific region from the BAM files (based on the search area above)
reads = impBamGAL(pathToFiles, 
                  searchArea, 
                  verbose = TRUE)

# Count the number of reads for the 2 different alleles at that position
countList = getAlleleCounts(reads, 
                           searchArea, 
                            verbose = TRUE)

# Save as a rds file for loading later if necessary
saveRDS(countList, 
        file = "/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE4/PD_rSNPs_254_het_counts_THREP1_ATACseq.rds")

# Load the new results of the 254 rSNPs
countList = read_rds("/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE4/PD_rSNPs_254_het_counts_THREP1_ATACseq.rds")

# Make an ASEset object to be able to plot
countList.ASE = ASEsetFromCountList(searchArea, countList)

# And plot one graph per pdf file
for (i in 1:length(countList.ASE)){
  pdf(paste0("/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE4/20241111_AlellicImbalance_254rSNPs_", 
             rownames(countList.ASE[i]),
             ".pdf"))
  barplot(countList.ASE[i])
  dev.off()
}

# Now check how many are truly heterozygous thanks to the whole genome sequencing data (WGS)
# Use table browser to retrieve according to this link http://genome.ucsc.edu/FAQ/FAQreleases.html#snpConversion
searchArea %>% 
  as_tibble() %>% 
  dplyr::select(-start) %>% # Not the real start when converting a GRanges to a tibble 
  mutate(start = end - 1) %>% 
  dplyr::select(seqnames, start, end, name) %>% 
  dplyr::rename(rSNPs = name) %>% 
  inner_join(PD_rSNPs_254_rsID.filt,
             by = "rSNPs") %>% 
  dplyr::select(name) %>% 
  write_delim(.,
              "/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE4/rsIDs_254PDrSNPs.txt",
              delim = "\t",
              col_names = FALSE)

# Load now the coordinates that are in hg19
PD_rSNPs_254_hg19 = read_delim("/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE4/rsIDs_254PDrSNPs_hg19_coord.txt",
                               delim = "\t",
                               col_names = TRUE)

# Only keep SNPs that are on canonical chromosomes
PD_rSNPs_254_hg19.GR = PD_rSNPs_254_hg19 %>% 
  dplyr::filter(!str_detect(`#chrom`, "_")) %>% 
  dplyr::select(`#chrom`:name) %>% 
  dplyr::rename(chr = `#chrom`) %>% 
  makeGRangesFromDataFrame(.,
                           keep.extra.columns = TRUE,
                           seqnames.field = "chr",
                           start.field = "chromStart",
                           end.field = "chromEnd",
                           starts.in.df.are.0based = TRUE)

# Load WGS data of TH REP1 mCHERRY cell line (hg19 genome version)
dat.SNPs.only.GR = read_rds("/Volumes/deborah.gerard/Documents/epifunc/SNPs_overlap/E19D017a78.merge.qc.SNP_INDEL.hg19.annotated.rds")

dat.SNPs.only.GR = dat.SNPs.only.GR %>% 
  dplyr::select(CHR:POS, REF:FILTER, E19D017a78) %>% 
  mutate(START = POS - 1) %>% 
  makeGRangesFromDataFrame(.,
                           keep.extra.columns = TRUE,
                           seqnames.field = "CHR",
                           start.field = "START",
                           end.field = "POS",
                           starts.in.df.are.0based = TRUE)

seqlevelsStyle(dat.SNPs.only.GR) = "UCSC"
seqlevelsStyle(dat.SNPs.only.GR)

# Overlap
hits = findOverlaps(PD_rSNPs_254_hg19.GR, dat.SNPs.only.GR)
rSNP.254_in_E19D017a78 = PD_rSNPs_254_hg19.GR[queryHits(hits)]
E19D017a78_w_rSNP.254 = dat.SNPs.only.GR[subjectHits(hits)]

mcols(rSNP.254_in_E19D017a78) = cbind(mcols(rSNP.254_in_E19D017a78),
                                 mcols(E19D017a78_w_rSNP.254))

rSNP.254_in_E19D017a78 = as.data.frame(rSNP.254_in_E19D017a78) %>% 
  as_tibble()

# Out of 254 PD-rSNPs, we have information about 242

# Save
write_delim(rSNP.254_in_E19D017a78,
            "/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE4/Genotype_of_254PDrSNPs_in_THREP1mCHERRY.txt",
            delim = "\t",
            col_names = TRUE)

# Load later
rSNP.254_in_E19D017a78 = read_delim("/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE4/Genotype_of_254PDrSNPs_in_THREP1mCHERRY.txt",
                                    delim = "\t",
                                    col_names = TRUE)

# Check how many are "unknown" (./.:.)
rSNP.254_in_E19D017a78 %>% 
  mutate(Genotype = gsub("\\:.*", "\\1", E19D017a78)) %>% 
  group_by(Genotype) %>% 
  summarise(GT.num = n()) # 54 PD-rSNPs are heterozygous in TH REP1 mCHERRY cell line, 24 are homozygous for the alternative allele and 164 are                                  unknown

# Check if the 54 PD-rSNPs that are heterozygous in TH REP1 mCHERRY cell line also lead to chromatin allelic imbalance
# rsIDs of the 54 heterozygous PD-rSNPs 
rSNP.254_in_E19D017a78_0.1 = rSNP.254_in_E19D017a78 %>% 
  mutate(Genotype = gsub("\\:.*", "\\1", E19D017a78),
         Genotype2 = gsub("\\|", "\\/", Genotype)) %>% 
  dplyr::filter(str_detect(Genotype2, "0/1")) %>% 
  #unite(col = "toInves", seqnames, end, sep = "_") %>% 
  dplyr::select(name)

# Get their coordinates
hetero_54_rSNP_in_E19D017a78.hg38 = searchArea %>% 
  as_tibble() %>% 
  dplyr::select(-start) %>% # Not the real start when converting a GRanges to a tibble 
  mutate(start = end - 1) %>% 
  dplyr::select(seqnames, start, end, name) %>% 
  dplyr::rename(rSNPs = name) %>% 
  inner_join(PD_rSNPs_254_rsID.filt,
             by = "rSNPs") %>% 
  dplyr::filter(name %in% rSNP.254_in_E19D017a78_0.1$name) %>% 
  unite(col = "coordToInves", seqnames, end, sep = "_")

# Extract the 54 heterozygous PD-rSNPs from thre ASet object
PDrSNPs_54_het = countList[intersect(names(countList), hetero_54_rSNP_in_E19D017a78.hg38$coordToInves)]

searchArea.0.1 = searchArea %>% 
  as_tibble() %>% 
  dplyr::filter(name %in% hetero_54_rSNP_in_E19D017a78.hg38$rSNPs)

searchArea.0.1 = searchArea.0.1 %>% 
  makeGRangesFromDataFrame(.,
                           keep.extra.columns = TRUE,
                           seqnames.field = "seqnames",
                           start.field = "start",
                           end.field = "end",
                           starts.in.df.are.0based = FALSE)

# And make it a ASet object back and proceed with a chi-squared test
PDrSNPs_54_het.ASE = ASEsetFromCountList(searchArea.0.1, PDrSNPs_54_het)
ref(PDrSNPs_54_het.ASE) = c("G", "T", "C")
PDrSNPs_54_het.ASE.chisq = as_tibble(chisq.test(PDrSNPs_54_het.ASE[,1:7], "+")) %>% 
  mutate(Sample = colnames(PDrSNPs_54_het.ASE)) %>% 
  dplyr::select(Sample, everything())

# Consider that a PD-rSNP lead to allelic imbalance if at least one sample (smNPC, mDAN-D15, mDAN-D30, mDAN-D50) shows allelic imbalance
PDrSNPs_54_het.ASE.chisq %>% 
  dplyr::filter(!str_detect(Sample, paste0(c("astrocyte", "neg"), collapse = "|"))) %>% 
  #pivot_longer(!Sample, names_to = "PDrSNPs", values_to = "pvalue") %>% 
  #group_by(Sample) %>% 
  #dplyr::filter(if_any(where(is.numeric), ~ . < 0.05))
  select_if(~any(. < 0.05)) %>% 
  mutate(AI_yes = ncol(.)) %>% 
  View()
# 46 among 54 PD-rSNPs show allelic imbalance

test = tibble(AI = c("yes", "no"),
       value = c(46, 8),
       percent = round((value/sum(value))*100, digits = 1))

pizza.AI = ggplot(test,
       aes(x = "", y = value, fill = AI)) +
  geom_bar(width = 1, stat = "identity", alpha = 0.5) +
    #theme_void() +
    scale_fill_manual(values = c("#4682BD", "#64A7D1")) +
    coord_polar(theta = "y", start = 2.125, clip = "off") +
    geom_text(aes(
        x = c(1.0, 1.2),
        y = c(21.5, 50),
        label = paste0(percent, "%")),
        size = 5.0, 
        color = "black",
        fontface = "bold") +
  theme_void() +
  theme(legend.position = "none")

plotGG(plot = pizza.AI, 
       x = 3.0, 
       y = 0.5, 
       width = 6.0, 
       height = 1.75,
       just = c("left", "top"), 
       default.units = "inches")
  
# And plot one graph per pdf file
for (i in 1:length(PDrSNPs_54_het.ASE)){
  pdf(paste0("/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE4/20241115_AlellicImbalance_54HeterorSNPs_",
             rownames(PDrSNPs_54_het.ASE[i]),
             ".pdf"))
  barplot(PDrSNPs_54_het.ASE[i])
  dev.off()
}

#### PANEL C - Manhattan plot #### 
# text C
plotText(label = "C", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 4.125, 
         y = 2.125, 
         just = "left", 
         default.units = "inches",
         fontface = "bold") 

# Load the Nalls et al. GWAS data
nalls_allSNPs.HG38.GR = readRDS("/home/vagrant/Manuscript_1/FIGURE1/nalls_allSNPs.HG38.GR.rds")
#nalls_allSNPs.HG38.GR = readRDS("/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE1/nalls_allSNPs.HG3#8.GR.rds")


# Load the rsIDS FOR THE 254 significant SNPs
rsID.254 = read_delim("/Volumes/deborah.gerard/Documents/epifunc/SNEEP2024/RESULTS_SEPTEMBER_2024/output/SNEEP_uniq_rSNPs.txt",
                     delim = "\t",
                     col_names = TRUE)

# Extract the chr and the position
rsID.254_2 = rsID.254 %>% 
  separate(col = "SNP_position", 
           into = c("chr", "pos"), 
           sep = ":",
           remove = FALSE) %>% 
  separate(col = "pos",
           into = c("start", "end"),
           sep = "-",
           remove = TRUE) %>% 
  unite(col = pos, 
        c("chr", "end"), 
        sep = "_", 
        remove = TRUE) %>% 
  dplyr::select(-start)

# Select the SNPs position from nalls et al and associate rsIDs
nalls.manH = nalls_allSNPs.HG38.GR %>% 
  as_tibble() %>% 
  dplyr::select(seqnames, base_pair_position, p) %>% 
  unite(col = "pos", seqnames, base_pair_position, sep = "_", remove = FALSE) %>%
  left_join(rsID.254_2, by = "pos") %>% 
  separate(pos, into = c("chrom", "pos"), sep = "_", remove = TRUE) %>% 
  dplyr::select(chrom, pos, p, SNP_position)

# Save
write_rds(nalls.manH,
          "/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE1/20241107_254_manH.rds")

# Load for later 
nalls.manH = read_rds("/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE1/20241107_254_manH.rds")

# Rename columns
nalls.manH = nalls.manH %>% 
  dplyr::rename(snp = SNP_position) %>% 
  mutate(pos = as.numeric(pos))

# Manhattan plot - highlight the 254 SNPs with rsID
# Get the rsID of the 254 rSNPs first
nalls.manH %>% 
  dplyr::filter(!is.na(snp),
                p <= 5e-08) %>% 
  dplyr::select(snp) %>% 
  write_delim(.,
              "/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE1/20241107_254PDrSNPs_coord.txt",
              col_names = FALSE,
              delim = "\t")

# Load the files with the associated rsIDs
PD_rSNPs_254_rsID = read_delim("/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE1/20241107_254PDrSNPs_rsID.txt",
                               delim = "\t",
                               col_names = TRUE)

# Only keep snv
PD_rSNPs_254_rsID.filt = PD_rSNPs_254_rsID %>% 
  filter(class == "snv") %>% 
  dplyr::select(`#chrom`:name) %>% 
  distinct(name, .keep_all = TRUE) %>% 
  unite(col = "tmp", `#chrom`, chromStart, sep = ":", remove = FALSE) %>% 
  unite(col = "rSNPs", tmp, chromEnd, sep = "-", remove = TRUE)

# Associated then the significant GWAS pvalue
snp_2_high = nalls.manH %>% 
  dplyr::filter(!is.na(snp),
                p <= 5e-08) %>% 
  dplyr::rename(rSNPs = snp) %>% 
  left_join(PD_rSNPs_254_rsID.filt,
            by = "rSNPs") %>% 
  dplyr::select(-rSNPs, -`#chrom`, -chromStart) %>% 
  dplyr::rename(snp = name)
  
# Get the full set of SNPs and associate rsID to the one I want to highlight
# snp_to_pl = nalls.manH %>% 
#   mutate(snp = NA_character_) %>% 
#   bind_rows(snp_2_high)

snp_to_pl = nalls.manH %>%
  mutate(snp = NA_character_) %>%
  bind_rows(snp_2_high) %>%
  mutate(pbis = -log10(p),
         chrom = factor(chrom,
                        levels = paste0("chr",
                                        1:22)),
         color = if_else(is.na(snp), "no_high", "high"),
         
         label = if_else(snp %in% c("rs1465922", "rs144814361"), snp, NA_character_))

highlight_colormap = c("no_high" = adjustcolor( "grey", 
                                                alpha.f = 0.2), 
                       "high" = "#6600FF")

no_high = manhattan_data_preprocess(snp_to_pl,
                                    pval.colname = "p",
                                    chr.colname = "chrom",
                                    pos.colname = "pos",
                                    highlight.colname = "color",
                                    highlight.col = highlight_colormap,
                                    signif = c(5e-08))
# Plot
snp.pl = manhattan_plot(x = no_high,
                        color.by.highlight = TRUE,
                        rescale = TRUE,
                        label.colname = "label") +
  theme(axis.text.x = element_text(size = 9,
                                 face = "bold",
                                 angle = 90),
        axis.text.y = element_text(size = 9,
                                   face = "bold"),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 11,
                                    face = "bold"))
 
# saveRDS(snp.pl,
#        "/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE2/Fig2C_ManhattanPlot.rds")
snp.pl = readRDS("/home/vagrant/Manuscript_1/FIGURE2/Fig2C_ManhattanPlot.rds")

# Place the plot
plotGG(plot = snp.pl, 
       x = 4.5, 
       y = 2.25, 
       width = 3.75, 
       height = 1.5,
       just = c("left", "top"), 
       default.units = "inches")

#### PANEL D - BAG3 locus #### 
# text D
plotText(label = "D", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 4.0, 
         just = "left", 
         default.units = "inches",
         fontface = "bold") 

# ATACseq signal BAG3
# path where bigwig files are
bw.path = "/home/vagrant/epifunc/"
#bw.path = "/Volumes/deborah.gerard/Documents/epifunc/"

# 1st bio replicate smNPC
bw.smNPC_1 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/smNPC_I.bw"),
                        chrom = "chr10",
                        chromstart = 119611305,
                        chromend = 119691505)

# 2nd bio replicate smNPC
bw.smNPC_2 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/smNPC_II.bw"),
                        chrom = "chr10",
                        chromstart = 119611305,
                        chromend = 119691505)

# 3rd bio replicate smNPC
bw.smNPC_3 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/smNPC_III.bw"),
                        chrom = "chr10",
                        chromstart = 119611305,
                        chromend = 119691505)

# 1st bio replicate TH positive neurons D15
bw.TH.pos_D15_1 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/D15_POS_I.bw"),
                        chrom = "chr10",
                        chromstart = 119611305,
                        chromend = 119691505)

# 2nd bio replicate TH positive neurons D15
bw.TH.pos_D15_2 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/D15_POS_II.bw"),
                        chrom = "chr10",
                        chromstart = 119611305,
                        chromend = 119691505)

# 3rd bio replicate TH positive neurons D15
bw.TH.pos_D15_3 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/D15_POS_III.bw"),
                        chrom = "chr10",
                        chromstart = 119611305,
                        chromend = 119691505)

# 1st bio replicate TH positive neurons D30
bw.TH.pos_D30_1 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/D30_POS_I.bw"),
                        chrom = "chr10",
                        chromstart = 119611305,
                        chromend = 119691505)

# 2nd bio replicate TH positive neurons D30
bw.TH.pos_D30_2 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/D30_POS_II.bw"),
                        chrom = "chr10",
                        chromstart = 119611305,
                        chromend = 119691505)

# 3rd bio replicate TH positive neurons D30
bw.TH.pos_D30_3 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/D30_POS_III.bw"),
                        chrom = "chr10",
                        chromstart = 119611305,
                        chromend = 119691505)

# 1st bio replicate TH positive neurons D50
bw.TH.pos_D50_1 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/HFFTHmCherry_D50_possort_S1.bw"),
                        chrom = "chr10",
                        chromstart = 119611305,
                        chromend = 119691505)

# 2nd bio replicate TH positive neurons D50
bw.TH.pos_D50_2 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/HFFTHmCherry_D50_possort_S3.bw"),
                        chrom = "chr10",
                        chromstart = 119611305,
                        chromend = 119691505)

# 3rd bio replicate TH positive neurons D50
bw.TH.pos_D50_3 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/HFFTHmCherry_D50_possort_S4.bw"),
                        chrom = "chr10",
                        chromstart = 119611305,
                        chromend = 119691505)

# Add a scale next to the bigwig files - check the maximum to choose
scale.BAG3.max = max(c(bw.smNPC_1$score,
                       bw.smNPC_2$score,
                       bw.smNPC_3$score,
                       bw.TH.pos_D15_1$score,
                       bw.TH.pos_D15_2$score,
                       bw.TH.pos_D15_3$score,
                       bw.TH.pos_D30_1$score,
                       bw.TH.pos_D30_2$score,
                       bw.TH.pos_D30_3$score,
                       bw.TH.pos_D50_1$score,
                       bw.TH.pos_D50_2$score,
                       bw.TH.pos_D50_3$score))

print(scale.BAG3.max)

# Define parameters for the regions
region.p.BAG3 = pgParams(chrom = "chr10",
                         chromstart = 119611405,
                         chromend = 119691405,
                         assembly = "hg38",
                         range = c(0, scale.BAG3.max))

# Add the genomic label
plotGenomeLabel(
  chrom = "chr10",
  chromstart = 119611405,
  chromend = 119691405,
  assembly = "hg38",
  x = 0.5,
  y = 4.125,
  length = 4.0,
  default.units = "inches",
  scale = "Mb")

# ATACseq signal smNPC_I and scale
ATAC_smNPC_I = plotSignal(
  data = bw.smNPC_1, 
  params = region.p.BAG3,
  fill = "#D3D3D3", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 4.375, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal smNPC_II
ATAC_smNPC_II = plotSignal(
  data = bw.smNPC_2, 
  params = region.p.BAG3,
  fill = "#C0C0C0", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 4.375, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal smNPC_III
ATAC_smNPC_III = plotSignal(
  data = bw.smNPC_3, 
  params = region.p.BAG3,
  fill = "#A9A9A9", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 4.375, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = ATAC_smNPC_III, 
          at = c(0, scale.BAG3.max),
          fontsize = 6)

# Add the samples name
plotText(label = "smNPC", 
         fontsize = 6, 
         fontcolor = "#A9A9A9",
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 4.5, 
         just = c("left", "top"),
         default.units = "inches",
         fontface = "bold")

# ATACseq signal TH positive neurons D15 I
ATAC_TH.pos_D15_I = plotSignal(
  data = bw.TH.pos_D15_1, 
  params = region.p.BAG3,
  fill = "#B22222", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 4.675, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal TH positive neurons D15 II
ATAC_TH.pos_D15_II = plotSignal(
  data = bw.TH.pos_D15_2, 
  params = region.p.BAG3,
  fill = "#A52A2A", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 4.675, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal TH positive neurons D15 III
ATAC_TH.pos_D15_III = plotSignal(
  data = bw.TH.pos_D15_3, 
  params = region.p.BAG3,
  fill = "#8B0000", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 4.675, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = ATAC_TH.pos_D15_III, 
          at = c(0, scale.BAG3.max),
          fontsize = 6)

# Annotate sample name
plotText(label = "mDAN D15", 
         fontsize = 6, 
         fontfamily = "Helvetica",
         fontcolor = "#B22222",
         x = 0.25, 
         y = 4.75, 
         just = c("left", "top"),
         default.units = "inches",
         fontface = "bold")

# ATACseq signal TH positive neurons D30 I
ATAC_TH.pos_D30_I = plotSignal(
  data = bw.TH.pos_D30_1, 
  params = region.p.BAG3,
  fill = "#FF6347", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 4.975, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal TH positive neurons D30 II
ATAC_TH.pos_D30_II = plotSignal(
  data = bw.TH.pos_D30_2, 
  params = region.p.BAG3,
  fill = "#FF0000", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 4.975, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal TH positive neurons D30 III
ATAC_TH.pos_D30_III = plotSignal(
  data = bw.TH.pos_D30_3, 
  params = region.p.BAG3,
  fill = "#DC143C", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 4.975,
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = ATAC_TH.pos_D30_III, 
          at = c(0, scale.BAG3.max),
          fontsize = 6)

# Annotate sample name
plotText(label = "mDAN D30", 
         fontsize = 6, 
         fontfamily = "Helvetica",
         fontcolor = "#DC143C",
         x = 0.25, 
         y = 5.0625, 
         just = c("left", "top"),
         default.units = "inches",
         fontface = "bold")

# ATACseq signal TH positive neurons D50 I
ATAC_TH.pos_D50_I = plotSignal(
  data = bw.TH.pos_D50_1, 
  params = region.p.BAG3,
  fill = "#FFA07A", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 5.275, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal TH positive neurons D50 II
ATAC_TH.pos_D50_II = plotSignal(
  data = bw.TH.pos_D50_2, 
  params = region.p.BAG3,
  fill = "#FA8072", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 5.275, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal TH positive neurons D50 III
ATAC_TH.pos_D50_III = plotSignal(
  data = bw.TH.pos_D50_3, 
  params = region.p.BAG3,
  fill = "#E9967A", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 5.275,
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = ATAC_TH.pos_D50_III, 
          at = c(0, scale.BAG3.max),
          fontsize = 6)

# Annotate sample name
plotText(label = "mDAN D50", 
         fontsize = 6, 
         fontfamily = "Helvetica",
         fontcolor = "#E9967A",
         x = 0.25, 
         y = 5.375, 
         just = c("left", "top"),
         default.units = "inches",
         fontface = "bold")

# Gene track
plotGenes(chrom = "chr10",
  chromstart = 119611405,
  chromend = 119691405,
  assembly = assembly(Genome = "hg38refGene", 
                      TxDb = "TxDb.Hsapiens.UCSC.hg38.refGene", 
                      OrgDb = "org.Hs.eg.db"),
  fontcolor = "black",
  fill = "black",
  x = 0.5,
  y = 5.625,
  width = 4.0,
  height = 0.375,
  just = c("left", "top"),
  default.units = "inches")

#### PANEL E - BAG3 and LHX1 expression (RNAseq)
# text E
plotText(label = "E", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 4.75, 
         y = 4.0, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# Load RPKM matrix obtained from Borja and rename columns
# dat.RPKM = read_delim("/home/vagrant/epifunc/RNAseq/RPKM_genename",
#                       delim = "\t", 
#                       col_names = TRUE)
# 
# dat.RPKM = dat.RPKM %>% 
#   dplyr::select(gene_id, gene_name, everything())

# RPKM data - Remove the negative sorted neurons and astrocytes
# dat.RPKM.filt = dat.RPKM %>% 
#   dplyr::select(!contains("negsort"), -starts_with("ASTRO")) %>% 
#   pivot_longer(cols = ends_with("bam"), names_to = "Sample", values_to = "RPKM") %>% 
#   mutate(Cond = case_when(str_detect(Sample, "D15_possort") ~ "Positively sorted neurons D15",
#                          str_detect(Sample, "D30_possort") ~ "Positively sorted neurons D30",
#                          str_detect(Sample, "D50_possort") ~ "Positively sorted neurons D50",
#                          TRUE ~ "smNPCs"),
#          Cond = factor(Cond, levels = c("smNPCs", "Positively sorted neurons D15",
#                                   "Positively sorted neurons D30", 
#                                   "Positively sorted neurons D50")),
#          gene_id = gsub("\\..*", "", gene_id))

# Save the matrix of expression
# write_rds(dat.RPKM.filt,
#           "/home/vagrant/Manuscript_1/FIGURE2/mat_RPKM.rds")

# Load it for making the plots
dat.RPKM.filt = read_rds("/home/vagrant/Manuscript_1/FIGURE2/mat_RPKM.rds")
#dat.RPKM.filt = read_rds("/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE2/mat_RPKM.rds")

# Expression for BAG3 and LHX1
BAG3.LHX1.exp = ggboxplot(dat.RPKM.filt %>% 
                            dplyr::filter(gene_name %in% c("BAG3", "LHX1")),
                          x = "Cond", 
                          y = "RPKM", 
                          add = "jitter", 
                          fill = "Cond", 
                          palette = c("#A9A9A9", "#B22222", "#DC143C", "#E9967A"),
                          facet.by = "gene_name",
                          xlab = "", 
                          ylab = "Reads Per Kilobase Million (RPKM)") +
  scale_x_discrete(labels = c("smNPC",
                              expression("mDAN D15"),
                              expression("mDAN D30"),
                              expression("mDAN D50"))) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 90))
  

# Place the plot
plotGG(plot = BAG3.LHX1.exp, 
       x = 5.0, 
       y = 4.125, 
       width = 3, 
       height = 3.5,
       just = c("left", "top"), 
       default.units = "inches")

# Add the SNP position that zoom in the TFBS
SNP_param = pgParams(chrom = "chr10",
                     chromstart = 119651405,
                     chromend = 119651405,
                     assembly = "hg38")

annoHighlight(plot = ATAC_smNPC_I, 
              params = SNP_param,
              fill = "#404788FF",
              y = 4.5, 
              height = 1.5, 
              just = c("left", "top"), 
              default.units = "inches")

annoZoomLines(plot = ATAC_smNPC_I, 
              params = SNP_param,
              y0 = 6.0, 
              x1 = c(2.25, 2.75), 
              y1 = 6.125, 
              default.units = "inches")

# Add the transcription factor binding motif (TFBS) for LHX1
pfm.LHX1 = getMatrixByID(JASPAR2020, 
                         ID = "MA1518.1")
LHX1 = new.env()

LHX1$LHX1 = pfm.LHX1@profileMatrix

LHX1 = as.list(LHX1)

# Remove the nucleotide position
LHX1_TFBS = ggseqlogo(LHX1) +
  annotate("rect", 
           xmin = 4.5, 
           xmax = 5.5, 
           ymin = -0.05, 
           ymax = 2.5, 
           alpha = .1, 
           col = "black", 
           fill = "#404788FF") +
  theme(axis.text.x = element_blank())

# Place the plot
plotGG(plot = LHX1_TFBS, 
       x = 1.4925, 
       y = 5.9375, 
       width = 1.5, 
       height = 1.5,
       just = c("left", "top"), 
       default.units = "inches")

# text F
plotText(label = "F", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 7.75, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# # ATACseq signal IDUA
# # 1st bio replicate smNPC
# bw.IDUA.smNPC_1 = readBigwig(file = paste0(bw.path, 
#                                       "BIGWIG/smNPC_I.bw"),
#                         chrom = "chr4",
#                         chromstart = 947008,
#                         chromend = 1027208)
# 
# # 2nd bio replicate smNPC
# bw.IDUA.smNPC_2 = readBigwig(file = paste0(bw.path, 
#                                       "BIGWIG/smNPC_II.bw"),
#                         chrom = "chr4",
#                         chromstart = 947008,
#                         chromend = 1027208)
# 
# # 3rd bio replicate smNPC
# bw.IDUA.smNPC_3 = readBigwig(file = paste0(bw.path, 
#                                       "BIGWIG/smNPC_III.bw"),
#                         chrom = "chr4",
#                         chromstart = 947008,
#                         chromend = 1027208)
# 
# # 1st bio replicate TH positive neurons D15
# bw.IDUA.TH.pos_D15_1 = readBigwig(file = paste0(bw.path, 
#                                       "BIGWIG/D15_POS_I.bw"),
#                         chrom = "chr4",
#                         chromstart = 947008,
#                         chromend = 1027208)
# 
# # 2nd bio replicate TH positive neurons D15
# bw.IDUA.TH.pos_D15_2 = readBigwig(file = paste0(bw.path, 
#                                       "BIGWIG/D15_POS_II.bw"),
#                         chrom = "chr4",
#                         chromstart = 947008,
#                         chromend = 1027208)
# 
# # 3rd bio replicate TH positive neurons D15
# bw.IDUA.TH.pos_D15_3 = readBigwig(file = paste0(bw.path, 
#                                       "BIGWIG/D15_POS_III.bw"),
#                         chrom = "chr4",
#                         chromstart = 947008,
#                         chromend = 1027208)
# 
# # 1st bio replicate TH positive neurons D30
# bw.IDUA.TH.pos_D30_1 = readBigwig(file = paste0(bw.path, 
#                                       "BIGWIG/D30_POS_I.bw"),
#                         chrom = "chr4",
#                         chromstart = 947008,
#                         chromend = 1027208)
# 
# # 2nd bio replicate TH positive neurons D30
# bw.IDUA.TH.pos_D30_2 = readBigwig(file = paste0(bw.path, 
#                                       "BIGWIG/D30_POS_II.bw"),
#                         chrom = "chr4",
#                         chromstart = 947008,
#                         chromend = 1027208)
# 
# # 3rd bio replicate TH positive neurons D30
# bw.IDUA.TH.pos_D30_3 = readBigwig(file = paste0(bw.path, 
#                                       "BIGWIG/D30_POS_III.bw"),
#                         chrom = "chr4",
#                         chromstart = 947008,
#                         chromend = 1027208)
# 
# # 1st bio replicate TH positive neurons D50
# bw.IDUA.TH.pos_D50_1 = readBigwig(file = paste0(bw.path, 
#                                       "BIGWIG/HFFTHmCherry_D50_possort_S1.bw"),
#                         chrom = "chr4",
#                         chromstart = 947008,
#                         chromend = 1027208)
# 
# # 2nd bio replicate TH positive neurons D50
# bw.IDUA.TH.pos_D50_2 = readBigwig(file = paste0(bw.path, 
#                                       "BIGWIG/HFFTHmCherry_D50_possort_S3.bw"),
#                         chrom = "chr4",
#                         chromstart = 947008,
#                         chromend = 1027208)
# 
# # 3rd bio replicate TH positive neurons D50
# bw.IDUA.TH.pos_D50_3 = readBigwig(file = paste0(bw.path, 
#                                       "BIGWIG/HFFTHmCherry_D50_possort_S4.bw"),
#                         chrom = "chr4",
#                         chromstart = 947008,
#                         chromend = 1027208)
# 
# # Add a scale next to the bigwig files - check the maximum to choose
# scale.IDUA.max = max(c(bw.IDUA.smNPC_1$score,
#                        bw.IDUA.smNPC_2$score,
#                        bw.IDUA.smNPC_3$score,
#                        bw.IDUA.TH.pos_D15_1$score,
#                        bw.IDUA.TH.pos_D15_2$score,
#                        bw.IDUA.TH.pos_D15_3$score,
#                        bw.IDUA.TH.pos_D30_1$score,
#                        bw.IDUA.TH.pos_D30_2$score,
#                        bw.IDUA.TH.pos_D30_3$score,
#                        bw.IDUA.TH.pos_D50_1$score,
#                        bw.IDUA.TH.pos_D50_2$score,
#                        bw.IDUA.TH.pos_D50_3$score))
# 
# print(scale.IDUA.max)
# 
# # Define parameters for the regions
# region.p.IDUA = pgParams(chrom = "chr4",
#                          chromstart = 947108,
#                          chromend = 1027108,
#                          assembly = "hg38",
#                          range = c(0, scale.IDUA.max))
# 
# # Add the genomic label
# plotGenomeLabel(
#   chrom = "chr4",
#   chromstart = 947108,
#   chromend = 1027108,
#   assembly = "hg38",
#   x = 0.75,
#   y = 4.375,
#   length = 4.0,
#   default.units = "inches",
#   scale = "Mb")
# 
# # ATACseq signal smNPC_I
# ATAC_smNPC_I_IDUA = plotSignal(
#   data = bw.IDUA.smNPC_1, 
#   params = region.p.IDUA,
#   fill = "#D3D3D3", 
#   alpha = 0.7, 
#   linecolor = NA,
#   x = 0.75, 
#   y = 4.625, 
#   width = 4.0, 
#   height = 0.25,
#   just = c("left", "top"), 
#   default.units = "inches")
# 
# annoYaxis(plot = ATAC_smNPC_I_IDUA, 
#           at = c(0, scale.IDUA.max),
#           fontsize = 6)
# 
# # ATACseq signal smNPC_II
# ATAC_smNPC_II_IDUA = plotSignal(
#   data = bw.IDUA.smNPC_2, 
#   params = region.p.IDUA,
#   fill = "#C0C0C0", 
#   alpha = 0.7, 
#   linecolor = NA,
#   x = 0.75, 
#   y = 4.625, 
#   width = 4.0, 
#   height = 0.25,
#   just = c("left", "top"), 
#   default.units = "inches")
# 
# annoYaxis(plot = ATAC_smNPC_II_IDUA, 
#           at = c(0, scale.IDUA.max),
#           fontsize = 6)
# 
# # Add the SNP position that zoom in the TFBS
# SNP_param_IDUA = pgParams(chrom = "chr4",
#                      chromstart = 987108,
#                      chromend = 987108,
#                      assembly = "hg38")
# 
# annoHighlight(plot = ATAC_smNPC_I_IDUA, 
#               params = SNP_param_IDUA,
#               fill = "#404788FF",
#               y = 4.75, 
#               height = 1.875, 
#               just = c("left", "top"), 
#               default.units = "inches")
# 
# annoZoomLines(plot = ATAC_smNPC_I_IDUA, 
#               params = SNP_param_IDUA,
#               y0 = 6.625, 
#               x1 = c(2.25, 3.25), 
#               y1 = 6.75, 
#               default.units = "inches")
# 
# # ATACseq signal smNPC_III
# ATAC_smNPC_III_IDUA = plotSignal(
#   data = bw.IDUA.smNPC_3, 
#   params = region.p.IDUA,
#   fill = "#A9A9A9", 
#   alpha = 0.7, 
#   linecolor = NA,
#   x = 0.75, 
#   y = 4.625, 
#   width = 4.0, 
#   height = 0.25,
#   just = c("left", "top"), 
#   default.units = "inches")
# 
# annoYaxis(plot = ATAC_smNPC_III_IDUA, 
#           at = c(0, scale.IDUA.max),
#           fontsize = 6)
# 
# # ATACseq signal TH positive neurons D15 I
# ATAC_TH.pos_D15_I_IDUA = plotSignal(
#   data = bw.IDUA.TH.pos_D15_1, 
#   params = region.p.IDUA,
#   fill = "#B22222", 
#   alpha = 0.7, 
#   linecolor = NA,
#   x = 0.75, 
#   y = 5.0, 
#   width = 4.0, 
#   height = 0.25,
#   just = c("left", "top"), 
#   default.units = "inches")
# 
# annoYaxis(plot = ATAC_TH.pos_D15_I_IDUA, 
#           at = c(0, scale.IDUA.max),
#           fontsize = 6)
# 
# # ATACseq signal TH positive neurons D15 II
# ATAC_TH.pos_D15_II_IDUA = plotSignal(
#   data = bw.IDUA.TH.pos_D15_2, 
#   params = region.p.IDUA,
#   fill = "#A52A2A", 
#   alpha = 0.7, 
#   linecolor = NA,
#   x = 0.75, 
#   y = 5.0, 
#   width = 4.0, 
#   height = 0.25,
#   just = c("left", "top"), 
#   default.units = "inches")
# 
# annoYaxis(plot = ATAC_TH.pos_D15_II_IDUA, 
#           at = c(0, scale.IDUA.max),
#           fontsize = 6)
# 
# # ATACseq signal TH positive neurons D15 III
# ATAC_TH.pos_D15_III_IDUA = plotSignal(
#   data = bw.IDUA.TH.pos_D15_3, 
#   params = region.p.IDUA,
#   fill = "#8B0000", 
#   alpha = 0.7, 
#   linecolor = NA,
#   x = 0.75, 
#   y = 5.0, 
#   width = 4.0, 
#   height = 0.25,
#   just = c("left", "top"), 
#   default.units = "inches")
# 
# annoYaxis(plot = ATAC_TH.pos_D15_III_IDUA, 
#           at = c(0, scale.IDUA.max),
#           fontsize = 6)
# 
# # ATACseq signal TH positive neurons D30 I
# ATAC_TH.pos_D30_I_IDUA = plotSignal(
#   data = bw.IDUA.TH.pos_D30_1, 
#   params = region.p.IDUA,
#   fill = "#FF6347", 
#   alpha = 0.7, 
#   linecolor = NA,
#   x = 0.75, 
#   y = 5.375, 
#   width = 4.0, 
#   height = 0.25,
#   just = c("left", "top"), 
#   default.units = "inches")
# 
# annoYaxis(plot = ATAC_TH.pos_D30_I_IDUA, 
#           at = c(0, scale.IDUA.max),
#           fontsize = 6)
# 
# # ATACseq signal TH positive neurons D30 II
# ATAC_TH.pos_D30_II_IDUA = plotSignal(
#   data = bw.IDUA.TH.pos_D30_2, 
#   params = region.p.IDUA,
#   fill = "#FF0000", 
#   alpha = 0.7, 
#   linecolor = NA,
#   x = 0.75, 
#   y = 5.375, 
#   width = 4.0, 
#   height = 0.25,
#   just = c("left", "top"), 
#   default.units = "inches")
# 
# annoYaxis(plot = ATAC_TH.pos_D30_II_IDUA, 
#           at = c(0, scale.IDUA.max),
#           fontsize = 6)
# 
# # ATACseq signal TH positive neurons D30 III
# ATAC_TH.pos_D30_III_IDUA = plotSignal(
#   data = bw.IDUA.TH.pos_D30_3, 
#   params = region.p.IDUA,
#   fill = "#DC143C", 
#   alpha = 0.7, 
#   linecolor = NA,
#   x = 0.75, 
#   y = 5.375, 
#   width = 4.0, 
#   height = 0.25,
#   just = c("left", "top"), 
#   default.units = "inches")
# 
# annoYaxis(plot = ATAC_TH.pos_D30_III_IDUA, 
#           at = c(0, scale.IDUA.max),
#           fontsize = 6)
# 
# # ATACseq signal TH positive neurons D50 I
# ATAC_TH.pos_D50_I_IDUA = plotSignal(
#   data = bw.IDUA.TH.pos_D50_1, 
#   params = region.p.IDUA,
#   fill = "#FFA07A", 
#   alpha = 0.7, 
#   linecolor = NA,
#   x = 0.75, 
#   y = 5.75, 
#   width = 4.0, 
#   height = 0.25,
#   just = c("left", "top"), 
#   default.units = "inches")
# 
# annoYaxis(plot = ATAC_TH.pos_D50_I_IDUA, 
#           at = c(0, scale.IDUA.max),
#           fontsize = 6)
# 
# # ATACseq signal TH positive neurons D50 II
# ATAC_TH.pos_D50_II_IDUA = plotSignal(
#   data = bw.IDUA.TH.pos_D50_2, 
#   params = region.p.IDUA,
#   fill = "#FA8072", 
#   alpha = 0.7, 
#   linecolor = NA,
#   x = 0.75, 
#   y = 5.75, 
#   width = 4.0, 
#   height = 0.25,
#   just = c("left", "top"), 
#   default.units = "inches")
# 
# annoYaxis(plot = ATAC_TH.pos_D50_II_IDUA, 
#           at = c(0, scale.IDUA.max),
#           fontsize = 6)
# 
# # ATACseq signal TH positive neurons D50 III
# ATAC_TH.pos_D50_III_IDUA = plotSignal(
#   data = bw.IDUA.TH.pos_D50_3, 
#   params = region.p.IDUA,
#   fill = "#E9967A", 
#   alpha = 0.7, 
#   linecolor = NA,
#   x = 0.75, 
#   y = 5.75, 
#   width = 4.0, 
#   height = 0.25,
#   just = c("left", "top"), 
#   default.units = "inches")
# 
# annoYaxis(plot = ATAC_TH.pos_D50_III_IDUA, 
#           at = c(0, scale.IDUA.max),
#           fontsize = 6)
# 
# # Gene track
# plotGenes(chrom = "chr4",
#   chromstart = 947108,
#   chromend = 1027108,
#   assembly = assembly(Genome = "hg38refGene", 
#                       TxDb = "TxDb.Hsapiens.UCSC.hg38.refGene", 
#                       OrgDb = "org.Hs.eg.db"),
#   fontcolor = "black",
#   fill = "black",
#   x = 0.75,
#   y = 6.125,
#   width = 4.0,
#   height = 0.375,
#   just = c("left", "top"),
#   default.units = "inches")
# 
# # Add the samples name
# plotText(label = "smNPC", 
#          fontsize = 6, 
#          fontcolor = "#A9A9A9",
#          fontfamily = "Helvetica",
#          x = 0.75, 
#          y = 4.625, 
#          just = c("left", "top"),
#          default.units = "inches",
#          fontface = "bold")
# 
# plotText(label = "mDAN D15", 
#          fontsize = 6, 
#          fontcolor = "#B22222",
#          fontfamily = "Helvetica",
#          x = 0.75, 
#          y = 5.0, 
#          just = c("left", "top"),
#          default.units = "inches",
#          fontface = "bold")
# 
# plotText(label = "mDAN D30", 
#          fontsize = 6, 
#          fontcolor = "#DC143C",
#          fontfamily = "Helvetica",
#          x = 0.75, 
#          y = 5.375, 
#          just = c("left", "top"),
#          default.units = "inches",
#          fontface = "bold")
# 
# plotText(label = "mDAN D50", 
#          fontsize = 6, 
#          fontcolor = "#E9967A",
#          fontfamily = "Helvetica",
#          x = 0.75, 
#          y = 5.75, 
#          just = c("left", "top"),
#          default.units = "inches",
#          fontface = "bold")
# 
# #### PANEL D - IDUA and ZBTB14 expression (RNAseq)
# # text D
# plotText(label = "D", 
#          fontsize = 12,
#          fontfamily = "Helvetica",
#          x = 5.0, 
#          y = 4.25, 
#          just = "left", 
#          default.units = "inches",
#          fontface = "bold")
# 
# IDUA.ZBTB14.exp = ggboxplot(dat.RPKM.filt %>% 
#                             dplyr::filter(gene_name %in% c("IDUA", "ZBTB14", "SLC26A1")) %>% 
#                               mutate(gene_name = factor(gene_name,
#                                                         levels = c("IDUA", "ZBTB14", "SLC26A1"))),
#                           x = "Cond", 
#                           y = "RPKM", 
#                           add = "jitter", 
#                           fill = "Cond", 
#                           palette = c("#A9A9A9", "#B22222", "#DC143C", "#E9967A"),
#                           facet.by = "gene_name",
#                           xlab = "", 
#                           ylab = "Reads Per Kilobase Million (RPKM)") +
#   scale_x_discrete(labels = c("smNPC",
#                               "mDAN D15",
#                               "mDAN D30",
#                               "mDAN D50")) +
#   theme(legend.position = "none",
#         axis.text.x = element_text(angle = 90))
#   
# 
# # Place the plot
# plotGG(plot = IDUA.ZBTB14.exp, 
#        x = 5.25, 
#        y = 4.25, 
#        width = 3, 
#        height = 3.5,
#        just = c("left", "top"), 
#        default.units = "inches")
# 
# # Add the transcription factor binding motif (TFBS) for ZBTB14
# pfm.ZBTB14 = getMatrixByID(JASPAR2020, 
#                          ID = "MA1650.1")
# ZBTB14 = new.env()
# 
# ZBTB14$ZBTB14 = pfm.ZBTB14@profileMatrix
# 
# ZBTB14 = as.list(ZBTB14)
# 
# # Remove the nucleotide position
# ZBTB14_TFBS = ggseqlogo(ZBTB14) +
#   annotate("rect", 
#            xmin = 8.5, 
#            xmax = 9.5, 
#            ymin = -0.05, 
#            ymax = 2.5, 
#            alpha = .1, 
#            col = "black", 
#            fill = "#404788FF") +
#   theme(axis.text.x = element_blank())
# 
# # Place the plot
# plotGG(plot = ZBTB14_TFBS, 
#        x = 1.6225, 
#        y = 6.625, 
#        width = 1.5, 
#        height = 1.5,
#        just = c("left", "top"), 
#        default.units = "inches")
# 
# # text E
# plotText(label = "E", 
#          fontsize = 12,
#          fontfamily = "Helvetica",
#          x = 0.25, 
#          y = 8.0, 
#          just = "left", 
#          default.units = "inches",
#          fontface = "bold")

# ATACseq signal SCARB2
# 1st bio replicate smNPC
bw.SCARB2.smNPC_1 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/smNPC_I.bw"),
                        chrom = "chr4",
                        chromstart = 76173617,
                        chromend = 76253817)

# 2nd bio replicate smNPC
bw.SCARB2.smNPC_2 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/smNPC_II.bw"),
                        chrom = "chr4",
                        chromstart = 76173617,
                        chromend = 76253817)

# 3rd bio replicate smNPC
bw.SCARB2.smNPC_3 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/smNPC_III.bw"),
                        chrom = "chr4",
                        chromstart = 76173617,
                        chromend = 76253817)

# 1st bio replicate TH positive neurons D15
bw.SCARB2.TH.pos_D15_1 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/D15_POS_I.bw"),
                        chrom = "chr4",
                        chromstart = 76173617,
                        chromend = 76253817)

# 2nd bio replicate TH positive neurons D15
bw.SCARB2.TH.pos_D15_2 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/D15_POS_II.bw"),
                        chrom = "chr4",
                        chromstart = 76173617,
                        chromend = 76253817)

# 3rd bio replicate TH positive neurons D15
bw.SCARB2.TH.pos_D15_3 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/D15_POS_III.bw"),
                        chrom = "chr4",
                        chromstart = 76173617,
                        chromend = 76253817)

# 1st bio replicate TH positive neurons D30
bw.SCARB2.TH.pos_D30_1 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/D30_POS_I.bw"),
                        chrom = "chr4",
                        chromstart = 76173617,
                        chromend = 76253817)

# 2nd bio replicate TH positive neurons D30
bw.SCARB2.TH.pos_D30_2 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/D30_POS_II.bw"),
                        chrom = "chr4",
                        chromstart = 76173617,
                        chromend = 76253817)

# 3rd bio replicate TH positive neurons D30
bw.SCARB2.TH.pos_D30_3 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/D30_POS_III.bw"),
                        chrom = "chr4",
                        chromstart = 76173617,
                        chromend = 76253817)

# 1st bio replicate TH positive neurons D50
bw.SCARB2.TH.pos_D50_1 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/HFFTHmCherry_D50_possort_S1.bw"),
                        chrom = "chr4",
                        chromstart = 76173617,
                        chromend = 76253817)

# 2nd bio replicate TH positive neurons D50
bw.SCARB2.TH.pos_D50_2 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/HFFTHmCherry_D50_possort_S3.bw"),
                        chrom = "chr4",
                        chromstart = 76173617,
                        chromend = 76253817)

# 3rd bio replicate TH positive neurons D50
bw.SCARB2.TH.pos_D50_3 = readBigwig(file = paste0(bw.path, 
                                      "BIGWIG/HFFTHmCherry_D50_possort_S4.bw"),
                        chrom = "chr4",
                        chromstart = 76173617,
                        chromend = 76253817)

# Add a scale next to the bigwig files - check the maximum to choose
scale.SCARB2.max = max(c(bw.SCARB2.smNPC_1$score,
                       bw.SCARB2.smNPC_2$score,
                       bw.SCARB2.smNPC_3$score,
                       bw.SCARB2.TH.pos_D15_1$score,
                       bw.SCARB2.TH.pos_D15_2$score,
                       bw.SCARB2.TH.pos_D15_3$score,
                       bw.SCARB2.TH.pos_D30_1$score,
                       bw.SCARB2.TH.pos_D30_2$score,
                       bw.SCARB2.TH.pos_D30_3$score,
                       bw.SCARB2.TH.pos_D50_1$score,
                       bw.SCARB2.TH.pos_D50_2$score,
                       bw.SCARB2.TH.pos_D50_3$score))

print(scale.SCARB2.max)

# Define parameters for the regions
region.p.SCARB2 = pgParams(chrom = "chr4",
                         chromstart = 76173717,
                         chromend = 76253717,
                         assembly = "hg38",
                         range = c(0, scale.SCARB2.max))

# Add the genomic label
plotGenomeLabel(
  chrom = "chr4",
  chromstart = 76173717,
  chromend = 76253717,
  assembly = "hg38",
  x = 0.5,
  y = 7.875,
  length = 4.0,
  default.units = "inches",
  scale = "Mb")

# ATACseq signal smNPC_I
ATAC_smNPC_I_SCARB2 = plotSignal(
  data = bw.SCARB2.smNPC_1, 
  params = region.p.SCARB2,
  fill = "#D3D3D3", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 8.125, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal smNPC_II
ATAC_smNPC_II_SCARB2 = plotSignal(
  data = bw.SCARB2.smNPC_2, 
  params = region.p.SCARB2,
  fill = "#C0C0C0", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 8.125, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal smNPC_III
ATAC_smNPC_III_SCARB2 = plotSignal(
  data = bw.SCARB2.smNPC_3, 
  params = region.p.SCARB2,
  fill = "#A9A9A9", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 8.125, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = ATAC_smNPC_III_SCARB2, 
          at = c(0, scale.SCARB2.max),
          fontsize = 6)

# Add the samples name
plotText(label = "smNPC", 
         fontsize = 6, 
         fontcolor = "#A9A9A9",
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 8.250, 
         just = c("left", "top"),
         default.units = "inches",
         fontface = "bold")

# ATACseq signal TH positive neurons D15 I
ATAC_TH.pos_D15_I_SCARB2 = plotSignal(
  data = bw.SCARB2.TH.pos_D15_1, 
  params = region.p.SCARB2,
  fill = "#B22222", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 8.4375, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal TH positive neurons D15 II
ATAC_TH.pos_D15_II_SCARB2 = plotSignal(
  data = bw.SCARB2.TH.pos_D15_2, 
  params = region.p.SCARB2,
  fill = "#A52A2A", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 8.4375, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal TH positive neurons D15 III
ATAC_TH.pos_D15_III_SCARB2 = plotSignal(
  data = bw.SCARB2.TH.pos_D15_3, 
  params = region.p.SCARB2,
  fill = "#8B0000", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 8.4375,  
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = ATAC_TH.pos_D15_III_SCARB2, 
          at = c(0, scale.SCARB2.max),
          fontsize = 6)

plotText(label = "mDAN D15", 
         fontsize = 6, 
         fontcolor = "#B22222",
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 8.5625, 
         just = c("left", "top"),
         default.units = "inches",
         fontface = "bold")

# ATACseq signal TH positive neurons D30 I
ATAC_TH.pos_D30_I_SCARB2 = plotSignal(
  data = bw.SCARB2.TH.pos_D30_1, 
  params = region.p.SCARB2,
  fill = "#FF6347", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 8.75, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal TH positive neurons D30 II
ATAC_TH.pos_D30_II_SCARB2 = plotSignal(
  data = bw.SCARB2.TH.pos_D30_2, 
  params = region.p.SCARB2,
  fill = "#FF0000", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 8.75, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal TH positive neurons D30 III
ATAC_TH.pos_D30_III_SCARB2 = plotSignal(
  data = bw.SCARB2.TH.pos_D30_3, 
  params = region.p.SCARB2,
  fill = "#DC143C", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 8.75, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = ATAC_TH.pos_D30_III_SCARB2, 
          at = c(0, scale.SCARB2.max),
          fontsize = 6)

plotText(label = "mDAN D30", 
         fontsize = 6, 
         fontcolor = "#DC143C",
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 8.875, 
         just = c("left", "top"),
         default.units = "inches",
         fontface = "bold")

# ATACseq signal TH positive neurons D50 I
ATAC_TH.pos_D50_I_SCARB2 = plotSignal(
  data = bw.SCARB2.TH.pos_D50_1, 
  params = region.p.SCARB2,
  fill = "#FFA07A", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 9.0625, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")


# ATACseq signal TH positive neurons D50 II
ATAC_TH.pos_D50_II_SCARB2 = plotSignal(
  data = bw.SCARB2.TH.pos_D50_2, 
  params = region.p.SCARB2,
  fill = "#FA8072", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 9.0625, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

# ATACseq signal TH positive neurons D50 III
ATAC_TH.pos_D50_III_SCARB2 = plotSignal(
  data = bw.SCARB2.TH.pos_D50_3, 
  params = region.p.SCARB2,
  fill = "#E9967A", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.5, 
  y = 9.0625, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = ATAC_TH.pos_D50_III_SCARB2, 
          at = c(0, scale.SCARB2.max),
          fontsize = 6)

plotText(label = "mDAN D50", 
         fontsize = 6, 
         fontcolor = "#E9967A",
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 9.125, 
         just = c("left", "top"),
         default.units = "inches",
         fontface = "bold")

# Gene track
plotGenes(
  chrom = "chr4",
  chromstart = 76173717,
  chromend = 76253717,
  assembly = assembly(Genome = "hg38refGene", 
                      TxDb = "TxDb.Hsapiens.UCSC.hg38.refGene", 
                      OrgDb = "org.Hs.eg.db"),
  fontcolor = "black",
  fill = "black",
  x = 0.5,
  y = 9.5,
  width = 4.0,
  height = 0.375,
  just = c("left", "top"),
  default.units = "inches")

# Add the SNP position that zoom in the TFBS
SNP_param_SCARB2 = pgParams(chrom = "chr4",
                     chromstart = 76213717,
                     chromend = 76213717,
                     assembly = "hg38")

annoHighlight(plot = ATAC_smNPC_I_SCARB2, 
              params = SNP_param_SCARB2,
              fill = "#404788FF",
              y = 8.25, 
              height = 1.75, 
              just = c("left", "top"), 
              default.units = "inches")

annoZoomLines(plot = ATAC_smNPC_I_SCARB2, 
              params = SNP_param_SCARB2,
              y0 = 10.0, 
              x1 = c(2.25, 2.75), 
              y1 = 10.125, 
              default.units = "inches")

# text G
plotText(label = "G", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 4.75, 
         y = 7.75, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

SCARB2.NR2C2.exp = ggboxplot(dat.RPKM.filt %>% 
                            dplyr::filter(gene_name %in% c("SCARB2", "NR2C2", "FAM47E")) %>% 
                              mutate(gene_name = factor(gene_name,
                                                        levels = c("SCARB2", "NR2C2", "FAM47E"))),
                          x = "Cond", 
                          y = "RPKM", 
                          add = "jitter", 
                          fill = "Cond", 
                          palette = c("#A9A9A9", "#B22222", "#DC143C", "#E9967A"),
                          facet.by = "gene_name",
                          xlab = "", 
                          ylab = "Reads Per Kilobase Million (RPKM)") +
  scale_x_discrete(labels = c("smNPC",
                              "mDAN D15",
                              "mDAN D30",
                              "mDAN D50")) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 90))
  

# Place the plot
plotGG(plot = SCARB2.NR2C2.exp, 
       x = 5.0, 
       y = 8.0, 
       width = 3, 
       height = 3.5,
       just = c("left", "top"), 
       default.units = "inches")

# Add the transcription factor binding motif (TFBS) for NR2C2
pfm.NR2C2 = getMatrixByID(JASPAR2020, 
                         ID = "MA1536.1")

# Reverse to position weight matrix as the SNP is on the reverse strand
pfm.NR2C2 = reverseComplement(pfm.NR2C2)

NR2C2 = new.env()

NR2C2$NR2C2 = pfm.NR2C2@profileMatrix

NR2C2 = as.list(NR2C2)

# Remove nucleotide
NR2C2_TFBS = ggseqlogo(NR2C2) +
  annotate("rect", 
           xmin = 3.5, 
           xmax = 4.5, 
           ymin = -0.05, 
           ymax = 2.0, 
           alpha = .1, 
           col = "black", 
           fill = "#404788FF") +
  theme(axis.text.x = element_blank()) 

# Place the plot
plotGG(plot = NR2C2_TFBS, 
       x = 1.5975, 
       y = 10.0, 
       width = 1.5, 
       height = 1.5,
       just = c("left", "top"), 
       default.units = "inches")

pageGuideHide()
dev.off()
```

### FIGURE 3
```{r fig3, eval = FALSE, echo = TRUE}
########## FIGURE 3 ##########
##############################
tiff("/home/vagrant/Manuscript_1/FIGURE3/FIGURE3.tiff",
     width = 8.27,
     height = 11.67,
     units = "in",
     res = 300,
     compression = "lzw")

# Create a A4 blank page
pageCreate(width = 8.27, 
           height = 11.67, 
           default.units = "inches",
           showGuides = TRUE)

# text Figure 3
plotText(label = "Figure 3", 
         fontsize = 14,
         fontface = "bold",
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 0.25, 
         just = "left", 
         default.units = "inches")

#### PANEL A - Scheme of transduction for knockdown and luciferase
# text A
plotText(label = "A", 
         fontsize = 12,
         fontface = "bold",
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 0.5, 
         just = "left", 
         default.units = "inches")

# Fig3A
fig3A = readPNG("/home/vagrant/Manuscript_1/FIGURE3/Fig3A_BIORENDER.png")
#fig3A = readPNG("/Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE3/Fig3A_BIORENDER.png")

plotRaster(image = fig3A,
           x = 0.25, 
           y = 1, 
           width = 8.0, 
           height = 1.0, 
           just = "left",
           interpolate = TRUE)

#### PANEL B - luciferase barplots with statistics
# text B
plotText(label = "B", 
         fontsize = 12,
         fontface = "bold",
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 2.0, 
         just = "left", 
         default.units = "inches")

Gluc_signal = tibble(Sample = c("BAG3-WT", "BAG3-MUT", "IDUA-WT", "IDUA-MUT", "SCARB2-WT", "SCARB2-MUT", "miniCMV", "Neg_CTRL"),
                     GLUC_n1_rep1 = c(319, 240, 154, 196, 379, 211, 431, 33),
                     GLUC_n1_rep2 = c(323, 237, 161, 205, 371, 224, 459, 33),
                     GLUC_n2_rep1 = c(201, 143, 121, 128, 278, 172, 253, 31),
                     GLUC_n2_rep2 = c(191, 149, 119, 126, 278, 162, 249, 31),
                     GLUC_n3_rep1 = c(249, 146, 136, 142, 223, 175, 212, 32),
                     GLUC_n3_rep2 = c(203, 152, 121, 125, 209, 168, 190, 33),
                     GLUC_n4_rep1 = c(348, 333, 220, 185, 396, 244, 319, 31),
                     GLUC_n4_rep2 = c(322, 306, 229, 186, 369, 227, 294, 32),
                     SEAP_n1_rep1 = c(72, 78, 54, 67, 99, 75, 128, 32),
                     SEAP_n1_rep2 = c(69, 70, 49, 66, 97, 79, 132, 29),
                     SEAP_n2_rep1 = c(50, 52, 45, 45, 76, 64, 80, 32),
                     SEAP_n2_rep2 = c(50, 52, 45, 51, 85, 63, 80, 32),
                     SEAP_n3_rep1 = c(52, 50, 43, 50, 66, 55, 68, 32),
                     SEAP_n3_rep2 = c(49, 53, 48, 46, 66, 62, 69, 31),
                     SEAP_n4_rep1 = c(64, 75, 58, 54, 103, 75, 111, 33),
                     SEAP_n4_rep2 = c(61, 74, 62, 56, 108, 77, 111, 31)) %>% 
  pivot_longer(!Sample, names_to = "Meas", values_to = "Abs_val") %>% 
  mutate(Meas2 = gsub("^([^_]*_[^_]*)_.*", "\\1", Meas)) %>% 
  group_by(Sample, Meas2) %>% 
  mutate(AVG = mean(Abs_val, na.rm = TRUE))

# Calculate ratio Gluc over SEAP
Gluc_signal_ratio = Gluc_signal %>% 
  dplyr::select(-Meas, -Abs_val) %>% 
  distinct(Meas2, .keep_all = TRUE) %>% 
  pivot_wider(names_from = "Meas2", values_from = "AVG") %>% 
  mutate(Ratio_within_bio_rep1 = GLUC_n1/SEAP_n1,
         Ratio_within_bio_rep2 = GLUC_n2/SEAP_n2,
         Ratio_within_bio_rep3 = GLUC_n3/SEAP_n3,
         Ratio_within_bio_rep4 = GLUC_n4/SEAP_n4) %>% 
  dplyr::select(Sample, Ratio_within_bio_rep1:Ratio_within_bio_rep4) %>% 
  pivot_longer(cols = c("Ratio_within_bio_rep1", "Ratio_within_bio_rep2", "Ratio_within_bio_rep3",
                        "Ratio_within_bio_rep4"),
               names_to = "Ratio",
               values_to = "value") %>% 
  mutate(Genes = gsub("(.*)-.*", "\\1", Sample))
 

# Calculate statistics - T-test
GLUC_stat = compare_means(value ~ Sample,
                          data = Gluc_signal_ratio %>% 
                            ungroup(),
                          method = "t.test")

# Plot for BAG3
GLUC_BAG3 = ggbarplot(Gluc_signal_ratio %>% 
                        dplyr::filter(Sample %in% c("BAG3-WT",
                                                     "BAG3-MUT",
                                                     "miniCMV",
                                                     "Neg_CTRL")), 
                      x = "Sample", 
                      y = "value",
                      width = 0.25,
                      add = c("mean_se", 
                              "jitter"),
                      fill = "grey",
                      xlab = "",
                      ylab = "Ratio Gluc/SEAP",
                      position = position_dodge(0.9)) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 90, hjust = 1.0, size = 9, face = "bold", family = "Helvetica"),
        axis.text.y = element_text(size = 9, face = "bold"),
        axis.title.y = element_text(size = 10, face = "bold")) +
  scale_y_continuous(expand = c(0, 0),
                     limits = c(0, 6.0)) +
  scale_x_discrete(labels = c("BAG3 other allele",
                              "BAG3 effect allele",
                              "Positive control",
                              "Negative control")) +
  stat_pvalue_manual(GLUC_stat %>% 
                       dplyr::filter(group1 == "BAG3-WT",
                              group2 == "BAG3-MUT"),
                     label = "p = {p.format}",
                     y.position = 5.6,
                     size = 3.0)

# Place the plot
plotGG(plot = GLUC_BAG3, 
       x = 0.50, 
       y = 2.25, 
       width = 3.5, 
       height = 4.25,
       just = c("left", "top"), 
       default.units = "inches")

# Plot for IDUA
# GLUC_IDUA = ggbarplot(Gluc_signal_ratio %>% 
#                         dplyr::filter(Sample %in% c("IDUA-WT",
#                                                     "IDUA-MUT",
#                                                     "miniCMV",
#                                                     "Neg_CTRL")), 
#                       x = "Sample", 
#                       y = "value",
#           width = 0.25,
#           add = c("mean_se", 
#                   "jitter"),
#           fill = "grey",
#           xlab = "",
#           ylab = "Ratio Gluc/SEAP",
#           position = position_dodge(0.9)) +
#   theme(legend.position = "none",
#         axis.text.x = element_text(angle = 90, hjust = 1.0, size = 9, face = "bold"),
#         axis.text.y = element_text(size = 9, face = "bold"),
#         axis.title.y = element_text(size = 10, face = "bold")) +
#   scale_x_discrete(labels = c("IDUA other allele",
#                               "IDUA effect allele",
#                               "Positive control",
#                               "Negative control")) +
#   scale_y_continuous(expand = c(0, 0),
#                      limits = c(0, 6.0)) +
#   stat_pvalue_manual(GLUC_stat %>% 
#                        dplyr::filter(group1 == "IDUA-WT",
#                               group2 == "IDUA-MUT"),
#                      label = "p = {p.format}",
#                      y.position = 4.0,
#                      size = 3.0)
# 
# # Place the plot
# plotGG(plot = GLUC_IDUA, 
#        x = 3.0, 
#        y = 2.25, 
#        width = 2, 
#        height = 4.25,
#        just = c("left", "top"), 
#        default.units = "inches")

# Plot for SCARB2
GLUC_SCARB2 = ggbarplot(Gluc_signal_ratio %>% 
                          dplyr::filter(Sample %in% c("SCARB2-WT",
                                                     "SCARB2-MUT",
                                                     "miniCMV",
                                                     "Neg_CTRL")), 
                        x = "Sample", 
                        y = "value",
          width = 0.25,
          add = c("mean_se", 
                  "jitter"),
          fill = "grey",
          xlab = "",
          ylab = "Ratio Gluc/SEAP",
          position = position_dodge(0.9)) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 90, hjust = 1.0, size = 9, face = "bold"),
        axis.text.y = element_text(size = 9, face = "bold"),
        axis.title.y = element_text(size = 10, face = "bold")) +
  scale_x_discrete(labels = c("SCARB2 other allele",
                              "SCARB2 effect allele",
                              "Positive control",
                              "Negative control")) +
  scale_y_continuous(expand = c(0, 0),
                     limits = c(0, 6.0)) +
  stat_pvalue_manual(GLUC_stat %>% 
                       dplyr::filter(group1 == "SCARB2-WT",
                              group2 == "SCARB2-MUT"),
                     label = "p = {p.format}",
                     y.position = 4.0,
                      size = 3.0)

# Place the plot
plotGG(plot = GLUC_SCARB2, 
       x = 4.5, 
       y = 2.25, 
       width = 3.5, 
       height = 4.25,
       just = c("left", "top"), 
       default.units = "inches")

#### PANEL C - knockdown barplots with statistics
# text C
plotText(label = "C", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 6.75, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# Knockdown of LHX1 and quantification of LHX1 and BAG3 expression 3 days post transduction
KD.BAG3.LHX1.3days = tibble(Sple = rep("shLHX1", 6),
                            Gene = c(rep("LHX1", 3),
                                     rep("BAG3", 3)),
                            Exp = c(0.411253539,
                                    0.359459217,
                                    0.367190337,
                                    0.972688658,
                                    0.965902853,
                                    0.88813445))

# Stats
KD.BAG3.LHX1.3days.stats = tibble(Sple = c(rep("shLHX1", 6),
                                           rep("shCTRL", 6)),
                                  Gene = rep(c(rep("LHX1", 3),
                                           rep("BAG3", 3)), 2),
                                  Exp = c(9.5167,
                                          8.39675,
                                          8.1855,
                                          9.13605,
                                          8.33795,
                                          8.76415,
                                          8.2348,
                                          6.92065,
                                          6.7401,
                                          9.0961,
                                          8.2879,
                                          8.593))

stat_LHX1.KD = compare_means(Exp ~ Sple,
                          data = KD.BAG3.LHX1.3days.stats,
                          group.by = "Gene",
                          method = "t.test",
                          paired = TRUE)
# Plot
pl.KD_BAG3_LHX1 = KD.BAG3.LHX1.3days %>% 
  mutate(Gene = factor(Gene, levels = c("LHX1", "BAG3"))) %>% 
  ggbarplot(.,
            x = "Gene",
            y = "Exp",
            fill = c("Gene"),
            position = position_dodge(0.9),
            palette = c("#440154FF", "#287C8EFF"),
            add = c("mean_se", "jitter"),
            xlab = "",
            ylab = "Relative expression to shSCRAMBLE",
            title = "shLHX1 - 3 days") +
  geom_hline(yintercept = 1.0, lty = "dashed", color = "black") +
  theme(axis.title.y = element_text(face = "bold", size = 10),
        axis.text.x = element_text(face = "bold.italic", size = 10, angle = 90, family = "Helvetica"),
        axis.text.y = element_text(face = "bold", size = 10),
        legend.position = "none",
        plot.title = element_text(face = "bold", size = 10, hjust = 0.5)) +
  scale_y_continuous(expand = c(0, 0),
                     limits = c(0, 1.8)) +
  stat_pvalue_manual(stat_LHX1.KD,
                     label = "p = {p.format}",
                     x = "Gene",
                     y.position = c(0.5, 1.2),
                     size = 3.0,
                     hide.ns = FALSE)

plotGG(plot = pl.KD_BAG3_LHX1, 
       x = 0.5, 
       y = 7.0, 
       width = 2.0, 
       height = 4.25,
       just = c("left", "top"), 
       default.units = "inches")

# Knockdown of ZBTB14 and quantification of ZBTB14 and IDUA expression 3 days post transduction
# KD.IDUA.ZBTB14.3days = tibble(Sple = rep("shZBTB14", 6),
#                             Gene = c(rep("ZBTB14", 3),
#                                      rep("IDUA", 3)),
#                             Exp = c(0.334134302,
#                                     0.316834205,
#                                     0.294206291,
#                                     0.663813558,
#                                     1.259979272,
#                                     1.774054093))
# 
# # Stats
# KD.IDUA.ZBTB14.3days.stats = tibble(Sple = c(rep("shZBTB14", 6),
#                                            rep("shCTRL", 6)),
#                                   Gene = rep(c(rep("ZBTB14", 3),
#                                            rep("IDUA", 3)), 2),
#                                   Exp = c(11.39125,
#                                           10.4813,
#                                           10.5062,
#                                           11.56275,
#                                           10.1455,
#                                           9.76485,
#                                           9.80975,
#                                           8.8231,
#                                           8.7411,
#                                           10.9716,
#                                           10.4789,
#                                           10.5919))
# 
# stat_ZBTB14.KD = compare_means(Exp ~ Sple,
#                           data = KD.IDUA.ZBTB14.3days.stats,
#                           group.by = "Gene",
#                           method = "t.test",
#                           paired = TRUE)
# 
# # Plot
# pl.KD_IDUA_ZBTB14 = KD.IDUA.ZBTB14.3days %>% 
#   mutate(Gene = factor(Gene, levels = c("ZBTB14", "IDUA"))) %>% 
#   ggbarplot(.,
#             x = "Gene",
#             y = "Exp",
#             fill = c("Gene"),
#             position = position_dodge(0.9),
#             palette = c("#440154FF", "#287C8EFF"),
#             add = c("mean_se", "jitter"),
#             xlab = "",
#             ylab = "Relative expression to shSCRAMBLE",
#             title = "shZBTB14 - 3 days") +
#   geom_hline(yintercept = 1.0, lty = "dashed", color = "black") +
#   theme(axis.title.y = element_text(face = "bold", size = 10),
#         axis.text.x = element_text(face = "bold.italic", size = 10, angle = 90, family = "Helvetica"),
#         axis.text.y = element_text(face = "bold", size = 10),
#         legend.position = "none",
#         plot.title = element_text(face = "bold", size = 10, hjust = 0.5)) +
#   scale_y_continuous(expand = c(0, 0),
#                      limits = c(0, 1.9)) +
#   stat_pvalue_manual(stat_ZBTB14.KD,
#                      label = "p = {p.format}",
#                      x = "Gene",
#                      y.position = c(0.5, 1.6),
#                      size = 3.0,
#                      hide.ns = FALSE)
# 
# plotGG(plot = pl.KD_IDUA_ZBTB14, 
#        x = 2.375, 
#        y = 7.0, 
#        width = 1.75, 
#        height = 4.25,
#        just = c("left", "top"), 
#        default.units = "inches")

# Knockdown of NR2C2 and quantification of NR2C2 and SCARB2 expression 3 days post transduction
KD.SCARB2.NR2C2.3days = tibble(Sple = rep("shNR2C2", 6),
                            Gene = c(rep("NR2C2", 3),
                                     rep("SCARB2", 3)),
                            Exp = c(0.345210918,
                                    0.512864636,
                                    0.548665969,
                                    1.239922544,
                                    1.130295504,
                                    1.170885498))

# Stats
KD.SCARB2.NR2C2.3days.stats = tibble(Sple = c(rep("shNR2C2", 6),
                                           rep("shCTRL", 6)),
                                  Gene = rep(c(rep("NR2C2", 3),
                                           rep("SCARB2", 3)), 2),
                                  Exp = c(9.43575,
                                          9.3043,
                                          9.263,
                                          5.6841,
                                          5.92295,
                                          5.9289,
                                          7.9013,
                                          8.34095,
                                          8.397,
                                          5.99435,
                                          6.09965,
                                          6.1565))

stat_NR2C2.KD = compare_means(Exp ~ Sple,
                          data = KD.SCARB2.NR2C2.3days.stats,
                          group.by = "Gene",
                          method = "t.test",
                          paired = TRUE)

# Plot
pl.KD_SCARB2_NR2C2 = KD.SCARB2.NR2C2.3days %>% 
  mutate(Gene = factor(Gene, levels = c("NR2C2", "SCARB2"))) %>% 
  ggbarplot(.,
            x = "Gene",
            y = "Exp",
            fill = c("Gene"),
            position = position_dodge(0.9),
            palette = c("#440154FF", "#287C8EFF"),
            add = c("mean_se", "jitter"),
            xlab = "",
            ylab = "Relative expression to shSCRAMBLE",
            title = "shNR2C2 - 3 days") +
  geom_hline(yintercept = 1.0, lty = "dashed", color = "black") +
  theme(axis.title.y = element_text(face = "bold", size = 10),
        axis.text.x = element_text(face = "bold.italic", size = 10, angle = 90, family = "Helvetica"),
        axis.text.y = element_text(face = "bold", size = 10),
        legend.position = "none",
        plot.title = element_text(face = "bold", size = 10, hjust = 0.5)) +
  scale_y_continuous(expand = c(0, 0),
                     limits = c(0, 1.8)) +
  stat_pvalue_manual(stat_NR2C2.KD,
                     label = "p = {p.format}",
                     x = "Gene",
                     y.position = c(0.65, 1.5),
                     size = 3.0,
                     hide.ns = FALSE)

plotGG(plot = pl.KD_SCARB2_NR2C2, 
       x = 3.0, 
       y = 7.0, 
       width = 2.0, 
       height = 4.25,
       just = c("left", "top"), 
       default.units = "inches")

# Knockdown of NR2C2 and quantification of NR2C2 and SCARB2 expression 6 days post transduction
KD.SCARB2.NR2C2.6days = tibble(Sple = rep("shNR2C2", 6),
                            Gene = c(rep("NR2C2", 3),
                                     rep("SCARB2", 3)),
                            Exp = c(0.662710186,
                                    0.456029798,
                                    0.787662775,
                                    1.410640142,
                                    1.425976296,
                                    1.332143962))

# Stats
KD.SCARB2.NR2C2.6days.stats = tibble(Sple = c(rep("shNR2C2", 6),
                                           rep("shCTRL", 6)),
                                  Gene = rep(c(rep("NR2C2", 3),
                                           rep("SCARB2", 3)), 2),
                                  Exp = c(8.34355,
                                          9.0917,
                                          8.32445,
                                          5.3595,
                                          5.19535,
                                          6.5266,
                                          7.75,
                                          7.9589,
                                          7.9801,
                                          5.85585,
                                          5.7073,
                                          6.94035))

stat_NR2C2.KD.6days = compare_means(Exp ~ Sple,
                          data = KD.SCARB2.NR2C2.6days.stats,
                          group.by = "Gene",
                          method = "t.test",
                          paired = TRUE)

# Plot
pl.KD_SCARB2_NR2C2_6days = KD.SCARB2.NR2C2.6days %>% 
  mutate(Gene = factor(Gene, levels = c("NR2C2", "SCARB2"))) %>% 
  ggbarplot(.,
            x = "Gene",
            y = "Exp",
            fill = c("Gene"),
            position = position_dodge(0.9),
            palette = c("#440154FF", "#287C8EFF"),
            add = c("mean_se", "jitter"),
            xlab = "",
            ylab = "Relative expression to shSCRAMBLE",
            title = "shNR2C2 - 6 days") +
  geom_hline(yintercept = 1.0, lty = "dashed", color = "black") +
  theme(axis.title.y = element_text(face = "bold", size = 10),
        axis.text.x = element_text(face = "bold.italic", size = 10, angle = 90, family = "Helvetica"),
        axis.text.y = element_text(face = "bold", size = 10),
        legend.position = "none",
        plot.title = element_text(face = "bold", size = 10, hjust = 0.5)) +
  scale_y_continuous(expand = c(0, 0),
                     limits = c(0, 1.8)) +
  stat_pvalue_manual(stat_NR2C2.KD.6days,
                     label = "p = {p.format}",
                     x = "Gene",
                     y.position = c(0.85, 1.5),
                     size = 3.0,
                     hide.ns = FALSE)

plotGG(plot = pl.KD_SCARB2_NR2C2_6days, 
       x = 5.5, 
       y = 7.0, 
       width = 2.0, 
       height = 4.25,
       just = c("left", "top"), 
       default.units = "inches")

pageGuideHide()
dev.off()
```

### FIGURE 4
Allelic imbalance at rs1465922 in ATACseq of differentiated TH REP1 cell line.  
```{bash, eval = FALSE, echo = TRUE}
# Copy the ATACseq BAM locally 
rsync -avzPhu --no-p iris-cluster://work/projects/epifunc/merged_bam/ /Volumes/deborah.gerard/Documents/Manuscript_1/FIGURE4/
```

```{r fig4, eval = FALSE, echo = TRUE}
########## FIGURE 4 ##########
##############################
# Save as TIFF, 300 dpi
tiff("/home/vagrant/Manuscript_1/FIGURE4/FIGURE4.tiff",
     width = 8.27,
     height = 11.67,
     units = "in",
     res = 300,
     compression = "lzw")

# Create a A4 blank page
pageCreate(width = 8.27, 
           height = 11.67, 
           default.units = "inches",
           showGuides = TRUE)

plotText(label = "Figure 4", 
         fontsize = 14,
         fontface = "bold",
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 0.25, 
         just = "left", 
         default.units = "inches")

#### PANEL A - Chromatin imbalance in TH REP1 mCHERRY cell line at rs1465922 
# text A
plotText(label = "A", 
         fontsize = 12,
         fontface = "bold",
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 0.5, 
         just = "left", 
         default.units = "inches")


# Chromatin imbalance values as calculated by the AllelicImbalance R package
countList = readRDS("/home/vagrant/Manuscript_1/FIGURE4/SCARB2_het_counts_THREP1_ATACseq.rds")

# Extract and reformat for positively sorted TH population
AI_ATAC = countList$chr4_76213717 %>% 
  as_tibble() %>% 
  mutate(Sample = rownames(countList$chr4_76213717)) %>% 
  pivot_longer(cols = c(A, C, G, T), names_to = "Base", values_to = "Reads") %>% 
  mutate(Sample.n = case_when(str_detect(Sample, "astrocytes") ~ "Astrocytes",
                              str_detect(Sample, "D15_neg") ~ "Negative DAN D15",
                              str_detect(Sample, "D15_pos") ~ "mDAN D15",
                              str_detect(Sample, "D30_pos") ~ "mDAN D30",
                              str_detect(Sample, "D50_neg") ~ "Negative DAN D50",
                              str_detect(Sample, "D50_pos") ~ "mDAN D50",
                              TRUE ~ "smNPCs")) %>% 
  dplyr::filter(!str_detect(Sample, "neg"),
                Base %in% c("A", "G")) %>% 
  mutate(Sample.n = factor(Sample.n,
                           levels = c("smNPCs",
                                      "mDAN D15",
                                      "mDAN D30",
                                      "mDAN D50",
                                      "Astrocytes")),
         Allele = case_when(Base == "A" ~ "PD allele",
                            TRUE ~ "Other allele"))
# Plot
chrom_AI = ggbarplot(AI_ATAC,
                     x = "Sample.n",
                     y = "Reads",
                     fill = "Allele",
                     position = position_dodge(0.7),
                     xlab = "",
                     ylab = "Reads",
                     label = format(c(paste0("p = ",
                                        0.3),
                          "",
                          paste0("p = ",
                                 0.3),
                          "",
                          paste0("p = ",
                                 0.003),
                          "",
                          paste0("p = ",
                                 0.0000003),
                          "",
                          paste0("p = ",
                                 0.2),
                          ""),
                          scientific = TRUE),
                     label.pos = "out",
                     lab.size = 3,
                     lab.vjust = 0.5,
                     lab.hjust = -0.5,
                     orientation = "horiz") +
  theme(legend.position = "right",
        axis.text.x = element_text(angle = 45, hjust = 1.0, face = "bold", size = 10),
        axis.text.y = element_text(face = "bold", size = 10),
        axis.title.y = element_text(face = "bold", size = 10),
        axis.title.x = element_text(face = "bold", size = 10),
        legend.text = element_text(face = "bold", size = 10),
        legend.title = element_text(face = "bold", size = 10),
        plot.title = element_text(hjust = 0.5)) +
  scale_fill_manual(values = c("#80A44F", "#427F96")) +
  scale_y_continuous(expand = c(0, 0),
                     limits = c(0, 330))

# Place the plot
plotGG(plot = chrom_AI, 
       x = 0.5, 
       y = 0.75, 
       width = 7, 
       height = 4,
       just = c("left", "top"), 
       default.units = "inches")

# text B
plotText(label = "B", 
         fontsize = 12,
         fontfamily = "Helvetica",
         x = 0.25, 
         y = 5.0, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# GTEX_SNPs_interest = read_delim("/Volumes/deborah.gerard/Documents/epifunc/eQTpLot/20230117_GTEX_data.csv",
#               delim = ";",
#              col_names = TRUE)

GTEX_SNPs_interest = read_delim("/home/vagrant/epifunc/eQTpLot/20230117_GTEX_data.csv",
               delim = ";",
               col_names = TRUE)

GTEX_SNPs_SCARB2_brain = GTEX_SNPs_interest %>% 
  mutate(GTEX_sig = if_else(P.Value < 0.05, "*", "")) %>% 
  dplyr::filter(str_detect(Tissue, "Brain"),
         Gene.Symbol == "SCARB2") %>% 
  mutate(SNP.Id = factor(SNP.Id,
                         levels = c("rs6825004",
                                    "rs7697073",
                                    "rs11547135",
                                    "rs6812193",
                                    "rs1465922")))

GTEX_eqtl = ggplot(data = GTEX_SNPs_SCARB2_brain, 
                   mapping = aes(x = SNP.Id,
                                 y = Tissue,
                                 fill = NES)) + 
  geom_tile() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, face = "bold", size = 10),
        axis.title.x = element_blank(),
        axis.text.y = element_text(face = "bold", size = 10),
        axis.title.y = element_blank(),
        legend.text = element_text(face = "bold", size = 10),
        legend.title = element_text(face = "bold", size = 10)) +
  scale_fill_gradient2(
    low = muted("blue"),
    mid = "white",
    high = muted("red"),
    midpoint = 0,
    name = "eQTL \neffect size") +
  geom_text(aes(label = GTEX_sig)) 

plotGG(plot = GTEX_eqtl, 
       x = 0.5, 
       y = 5.25, 
       width = 7, 
       height = 6,
       just = c("left", "top"), 
       default.units = "inches")


pageGuideHide()
dev.off()
```

### FIGURE 5
```{r fig5, eval = FALSE, echo = TRUE}
########## FIGURE 5 ##########
##############################
# Save as a PDF
pdf("/home/vagrant/Manuscript_1/FIGURE5/FIGURE5.pdf",
    width = 8.3, 
    height = 11.7)

# Save as TIFF, 300 ppi
tiff("/home/vagrant/Manuscript_1/FIGURE5/FIGURE5.tiff",
     width = 8.27,
     height = 11.67,
     units = "in",
     res = 300,
     compression = "lzw")

# Create a A4 blank page
pageCreate(width = 8.3, 
           height = 11.7, 
           default.units = "inches",
           showGuides = FALSE)

#### PANEL A - 
# text Figure 5
plotText(label = "Figure 5", 
         fontsize = 14,
         
         x = 0.25, 
         y = 0.25, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")
# text A
plotText(label = "A", 
         fontsize = 12,
         x = 0.25, 
         y = 0.5, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

GCase_signal = tibble(Sample = c("shSCARB2", "shNR2C2", "shSCRAMBLE"),
                     n1_rep1 = c(11577, 9485, 9903),
                     n1_rep2 = c(11329, 9685, 9574),
                     n2_rep1 = c(8979, 9443, 9489),
                     n2_rep2 = c(9489, 9440, 11333),
                     n3_rep1 = c(9209, 10130, 10353),
                     n3_rep2 = c(10104, 11158, 14640),
                     n4_rep1 = c(15255, 15551, 15611),
                     n4_rep2 = c(14964, 16078, 16352)) %>% 
  pivot_longer(!Sample, names_to = "Meas", values_to = "Abs_val") %>% 
  mutate(Meas2 = gsub("\\_.*", "\\1", Meas)) %>% 
  group_by(Sample, Meas2) %>% 
  mutate(AVG = mean(Abs_val, na.rm = TRUE)) %>% 
  ungroup() %>% 
  dplyr::select(-Meas, -Abs_val, -Meas2) %>%
  distinct(AVG, .keep_all = TRUE)
 
# Plot 
GCase.pl = ggbarplot(GCase_signal, 
                      x = "Sample", 
                      y = "AVG",
                      width = 0.25,
                      add = c("mean_se", 
                              "jitter"),
                      fill = "grey",
                      xlab = "",
                      ylab = "GCase absolute fluorescence value",
                      position = position_dodge(0.9)) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 90, hjust = 1.0, size = 9, face = "bold", family = "Helvetica"),
        axis.text.y = element_text(size = 9, face = "bold"),
        axis.title.y = element_text(size = 10, face = "bold")) +
  scale_y_continuous(expand = c(0, 0),
                     limits = c(0, 17000.0))

# Place the plot
plotGG(plot = GCase.pl, 
       x = 0.50, 
       y = 0.5, 
       width = 2, 
       height = 4.25,
       just = c("left", "top"), 
       default.units = "inches")

dev.off()







############ WILL BE SUPP #####

KD_TH_lev = tibble(Sple = rep("shNR2C2", 3),
                   Rep = 1:3,
                   days = "6 days",
                   Gene = "TH",
                   Exp = c(1.705920009,
                           2.196566231,
                           1.107610012))

#TH levels in knockdown of NR2C2 6 days#
TH_lev_KD = KD_TH_lev %>% 
  ggbarplot(.,
            x = "Gene",
            y = "Exp",
            fill = c("days"),
            facet.by = "Sple",
            position = position_dodge(0.9),
            palette = c("#287C8EFF"),
            add = c("mean_se"),
            xlab = "",
            ylab = "Relative expression to shSCRAMBLE") +
  geom_hline(yintercept = 1.0, lty = "dashed", color = "black") +
  theme(axis.title.y = element_text(face = "bold", size = 10),
        axis.text.x = element_text(face = "bold", size = 10, angle = 90),
        axis.text.y = element_text(face = "bold", size = 10),
        legend.position = "none")

plotGG(plot = TH_lev_KD, 
       x = 0.5, 
       y = 0.5, 
       width = 2, 
       height = 4.25,
       just = c("left", "top"), 
       default.units = "inches")

### DEXOMAG - 3 biological repliactes ###
GCase = read_excel("/Volumes/deborah.gerard/Documents/Wetlab/epifunc/DEXOMAG_shSCARB2_shNR2C2_shCTRL_N1_N2_N3/20231214_DEXOMAG_KD_SCARB2_NR2C2_SCRAMBLE_N1_N2_N3_ANALYSIS.xlsx")

GCase %>% 
  dplyr::select(GCase, `GCase MEAN`, ...10, ...11) %>% 
  drop_na() %>% 
  dplyr::rename(Sample = GCase,
                N1 = `GCase MEAN`,
                N2 = ...10,
                N3 = ...11) %>% 
  pivot_longer(N1:N3, names_to = "Rep.bio", values_to = "norm.fluo.value") %>% 
  mutate(norm.fluo.value = as.numeric(norm.fluo.value)) %>% 
  #dplyr::filter(Sample != "Whole Gcase NT") %>% 
  ggbarplot(.,
            x = "Sample",
            y = "norm.fluo.value",
            add = c("mean_se", "jitter"),
            xlab = "",
            ylab = "GCase activity (normalised fluorescence values)") +
  scale_x_discrete(labels = c("SCARB2 KD",
                              "NR2C2 KD",
                              "SCRAMBLE",
                              "Not transduced + dexomag",
                              "Not transduced - dexomag",
                              "Whole cell lysate")) +
  theme(axis.text.x = element_text(angle = 45, face = "bold", hjust = 1.0),
        axis.text.y = element_text(face = "bold"),
        axis.title.y = element_text(face = "bold")) +
  stat_compare_means(comparisons = list(c("shSCARB2", "shSCRAMBLE")),
                     method = "t.test",
                     size = 4,
                     label = "p.format") +
  stat_compare_means(comparisons = list(c("shNR2C2", "shSCRAMBLE")),
                     method = "t.test",
                     size = 4,
                     label = "p.format")
  

dev.off()
```

### SUPPLEMENTARY FIGURE 1
Plot genes of interest from the TH REP2 mCHERRY cell line RNAseq data to demonstrate that expression is similar.  
In addition, do the same for FOUNDIN-PD.
```{r suppFig1, eval = FALSE, echo = TRUE}
############################################
########## SUPPLEMENTARY FIGURE 1 ##########
############################################
# Save as TIFF, 300 dpi
tiff("/home/vagrant/Manuscript_1/SUPPLEMENTARY_FIGURE1/SUPPLEMENTARY_FIGURE1.tiff",
     width = 8.27,
     height = 9.0,
     units = "in",
     res = 300,
     compression = "lzw")

# Create a A4 blank page
pageCreate(width = 8.27, 
           height = 9.0, 
           default.units = "inches",
           showGuides = TRUE)

#### PANEL A - 
# text Supplementary Figure 1
plotText(label = "Supplementary Figure 1", 
         fontsize = 14,
         x = 0.25, 
         y = 0.25, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")
# text A
plotText(label = "A", 
         fontsize = 12,
         x = 0.25, 
         y = 0.5, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# Load data
dat.THREP2.FPKM = read_delim("/home/vagrant/epifunc/RNAseq/TH_REP2_mCHERRY/180604_HT-Rep2_fpkm.tsv",
                      delim = "\t", 
                      col_names = TRUE)

# Filter for genes of interest - BAG3 and LHX1 - IDUA, ZBTB14, SLC26A1 - SCARB2, NR2C2, FAM47E
BAG3.LHX1 = c("BAG3",
              "LHX1")

IDUA.ZBTB14 = c("IDUA",
                "ZBTB14",
                "SLC26A1")

SCARB2.NR2C2 = c("SCARB2",
                 "NR2C2",
                 "FAM47E")

# Reformat the matrix and take only positively sorted neurons
dat.THREP2.FPKM = dat.THREP2.FPKM %>% 
  pivot_longer(starts_with("TH"), 
               names_to = "Samples",
               values_to = "FPKM") %>% 
  #dplyr::filter(str_detect(Samples, paste("SmNPC", "pos", sep = "|"))) %>% 
  mutate(Cond = case_when(str_detect(Samples, "D15pos") ~ "Positively sorted neurons D15",
                          str_detect(Samples, "D15neg") ~ "Negatively sorted neurons D15",
                          str_detect(Samples, "D30pos") ~ "Positively sorted neurons D30",
                          str_detect(Samples, "D30neg") ~ "Negatively sorted neurons D30",
                          TRUE ~ "smNPCs"),
         Cond = factor(Cond, 
                       levels = c("smNPCs", 
                                  "Negatively sorted neurons D15",
                                  "Positively sorted neurons D15",
                                  "Negatively sorted neurons D30",
                                  "Positively sorted neurons D30")))

# Plot the expression of BAG3 and LHX1 in smNPC and mDAN
cdt = c("smNPCs",
        "Positively")

TH.REP2_BAG3.LHX1.exp = ggboxplot(dat.THREP2.FPKM %>% 
                            dplyr::filter(gene_name %in% BAG3.LHX1,
                                          str_detect(Cond, paste0(cdt, collapse = "|"))),
                          x = "Cond", 
                          y = "FPKM", 
                          add = "jitter", 
                          fill = "Cond", 
                          palette = c("#A9A9A9", "#B22222", "#DC143C"),
                          facet.by = "gene_name",
                          xlab = "", 
                          ylab = "Fragments Per Kilobase Million (FPKM)") +
  scale_x_discrete(labels = c("smNPC",
                              expression("mDAN D15"),
                              expression("mDAN D30"))) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 90))

# Place the plot  
plotGG(plot = TH.REP2_BAG3.LHX1.exp, 
       x = 0.5, 
       y = 0.625, 
       width = 3.5, 
       height = 3.5,
       just = c("left", "top"), 
       default.units = "inches")

# Plot the expression of IDUA and ZBTB14
# TH.REP2_IDUA.ZBTB14.exp = ggboxplot(dat.THREP2.FPKM %>% 
#                             dplyr::filter(gene_name %in% IDUA.ZBTB14) %>% 
#                               mutate(gene_name = factor(gene_name,
#                                                         levels = c("IDUA", "ZBTB14", "SLC26A1"))),
#                           x = "Cond", 
#                           y = "FPKM", 
#                           add = "jitter", 
#                           fill = "Cond", 
#                           palette = c("#A9A9A9", "#B22222", "#DC143C"),
#                           facet.by = "gene_name",
#                           xlab = "", 
#                           ylab = "Fragments Per Kilobase Million (FPKM)") +
#   scale_x_discrete(labels = c("smNPC",
#                               expression("mDAN D15"),
#                               expression("mDAN D30"))) +
#   theme(legend.position = "none",
#         axis.text.x = element_text(angle = 90))
# 
# # Place the plot  
# plotGG(plot = TH.REP2_IDUA.ZBTB14.exp, 
#        x = 2.75, 
#        y = 0.625, 
#        width = 2.5, 
#        height = 3.5,
#        just = c("left", "top"), 
#        default.units = "inches")

# Plot the expression of SCARB2 and NR2C2
TH.REP2_SCARB2.NR2C2.exp = ggboxplot(dat.THREP2.FPKM %>% 
                            dplyr::filter(gene_name %in% SCARB2.NR2C2,
                                          str_detect(Cond, paste0(cdt, collapse = "|"))) %>% 
                              mutate(gene_name = factor(gene_name,
                                                        levels = c("SCARB2", "NR2C2", "FAM47E"))),
                          x = "Cond", 
                          y = "FPKM", 
                          add = "jitter", 
                          fill = "Cond", 
                          palette = c("#A9A9A9", "#B22222", "#DC143C"),
                          facet.by = "gene_name",
                          xlab = "", 
                          ylab = "Fragments Per Kilobase Million (FPKM)") +
  scale_x_discrete(labels = c("smNPC",
                              expression("mDAN D15"),
                              expression("mDAN D30"))) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 90))

# Place the plot  
plotGG(plot = TH.REP2_SCARB2.NR2C2.exp, 
       x = 4.5, 
       y = 0.625, 
       width = 3.5, 
       height = 3.5,
       just = c("left", "top"), 
       default.units = "inches")

#### PANEL B - 
# text B
plotText(label = "B", 
         fontsize = 12,
         x = 0.25, 
         y = 4.5, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# FOUNDIN-PD data
# Load
FIPD = read_delim("/home/vagrant/epifunc/PPMI_data/FOUNDIN-PD_150.9_RNAB/aggregated_expression/cpmTable.tsv",
                  delim = "\t",
                  col_names = TRUE)

# Filter for genes of interest and get the ensembl and gene names
goi_ens = AnnotationDbi::select(org.Hs.eg.db, 
                 keys = c("BAG3",
                          "LHX1",
                          #"IDUA",
                          #"ZBTB14",
                          #"SLC26A1"
                          "SCARB2",
                          "NR2C2",
                          "FAM47E"), 
                 columns = c("SYMBOL",
                             "ENSEMBL"), 
                 keytype = "SYMBOL")

# 2 LHX1 ensembl IDs came back. Check on the Ensembl website which one is correct
goi_ens = goi_ens %>% 
  as_tibble() %>% 
  dplyr::filter(!ENSEMBL %in% c("ENSG00000274577",
                                "ENSG00000262446")) %>% 
  dplyr::rename(ENSEMBL.not.dot = ENSEMBL)

# Filter the matrix of expression for my genes of interest
FIPD.fin = FIPD %>% 
  dplyr::filter(str_detect(Geneid, paste(goi_ens$ENSEMBL.not.dot, collapse = "|"))) %>% 
  dplyr::rename(ENSEMBL = Geneid) %>% 
  mutate(ENSEMBL.not.dot = gsub("\\..*", "", ENSEMBL)) %>% 
  dplyr::select(ENSEMBL, ENSEMBL.not.dot, everything()) %>% 
  pivot_longer(cols = starts_with("RNAB"), names_to = "Samples", values_to = "CPM") %>% 
  mutate(Day = str_extract(Samples, "da+\\d+")) %>% 
  left_join(goi_ens, 
            by = "ENSEMBL.not.dot") %>% 
  dplyr::select(SYMBOL, Day, CPM)

# Anova
# BAG3.aov = FIPD.fin %>% 
#   dplyr::filter(SYMBOL %in% BAG3.LHX1) %>% 
#   group_by(SYMBOL) %>% 
#   anova_test(CPM ~ Day)

# Tukey post-hoc test
# BAG3.ph = FIPD.fin %>% 
#   dplyr::filter(SYMBOL %in% BAG3.LHX1) %>% 
#   group_by(SYMBOL) %>% 
#   tukey_hsd(CPM ~ Day) %>% 
#   add_xy_position(x = "Day")

# Plot expression of BAG3 and LHX1
FIPD.exp_BAG3.LHX1 = ggboxplot(FIPD.fin %>% 
                          dplyr::filter(SYMBOL %in% BAG3.LHX1),
                     x = "Day", 
                     y = "CPM", 
                     add = "jitter", 
                     fill = "Day", 
                     palette = c("#482677FF", "#2D708EFF", "#29AF7FFF"),
                     facet.by = "SYMBOL",
                     xlab = "", 
                     ylab = "Counts per Million (CPM)",
                     add.params = list(size = 0.1)) +
  scale_x_discrete(labels = c("Day 0",
                              "Day 25",
                              "Day 65")) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 90))
  # stat_pvalue_manual(BAG3.ph, 
  #                    hide.ns = TRUE,
  #                    label = "p.adj") +
  # labs(subtitle = get_test_label(BAG3.aov, 
  #                                detailed = FALSE))
  

# Place the plot  
plotGG(plot = FIPD.exp_BAG3.LHX1, 
       x = 0.5, 
       y = 4.75 ,
       width = 3.5, 
       height = 3.5,
       just = c("left", "top"), 
       default.units = "inches")

# Plot expression of IDUA and ZBTB14
# FIPD.exp_IDUA.ZBTB14 = ggboxplot(FIPD.fin %>% 
#                                    dplyr::filter(SYMBOL %in% IDUA.ZBTB14) %>% 
#                             mutate(SYMBOL = factor(SYMBOL, levels = c("IDUA", "ZBTB14", "SLC26A1"))),
#                      x = "Day", 
#                      y = "CPM", 
#                      add = "jitter", 
#                      fill = "Day", 
#                      palette = c("#482677FF", "#2D708EFF", "#29AF7FFF"),
#                      facet.by = "SYMBOL",
#                      xlab = "", 
#                      ylab = "Counts per Million (CPM)",
#                      add.params = list(size = 0.1)) +
#   scale_x_discrete(labels = c("Day 0",
#                               "Day 25",
#                               "Day 65")) +
#   theme(legend.position = "none",
#         axis.text.x = element_text(angle = 90))
  # stat_pvalue_manual(IDUA.ph, 
  #                    hide.ns = TRUE,
  #                    label = "p.adj") +
  # labs(subtitle = get_test_label(IDUA.aov, 
  #                                detailed = FALSE))

# Place the plot  
# plotGG(plot = FIPD.exp_IDUA.ZBTB14, 
#        x = 3.125, 
#        y = 4.75 ,
#        width = 2.5, 
#        height = 3.5,
#        just = c("left", "top"), 
#        default.units = "inches")

# Plot expression of SCARB2 alone
FIPD.exp_SCARB2 = ggboxplot(FIPD.fin %>% 
                                   dplyr::filter(SYMBOL %in% "SCARB2"),
                     x = "Day", 
                     y = "CPM", 
                     add = "jitter", 
                     fill = "Day", 
                     palette = c("#482677FF", "#2D708EFF", "#29AF7FFF"),
                     facet.by = "SYMBOL",
                     xlab = "", 
                     ylab = "Counts per Million (CPM)",
                     add.params = list(size = 0.1)) +
  scale_x_discrete(labels = c("Day 0",
                              "Day 25",
                              "Day 65")) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 90))
  # stat_pvalue_manual(SCARB2.ph, 
  #                    hide.ns = TRUE,
  #                    label = "p.adj") +
  # labs(subtitle = get_test_label(SCARB2.aov, 
  #                                detailed = FALSE))

# Place the plot  
plotGG(plot = FIPD.exp_SCARB2, 
       x = 4.25, 
       y = 4.75 ,
       width = 1.5, 
       height = 3.5,
       just = c("left", "top"), 
       default.units = "inches")

# Plot expression of NR2C2 and FAM47E
FIPD.exp_NR2C2.FAM47E = ggboxplot(FIPD.fin %>% 
                                   dplyr::filter(SYMBOL %in% c("NR2C2", "FAM47E")) %>% 
                            mutate(SYMBOL = factor(SYMBOL, levels = c("NR2C2", "FAM47E"))),
                     x = "Day", 
                     y = "CPM", 
                     add = "jitter", 
                     fill = "Day", 
                     palette = c("#482677FF", "#2D708EFF", "#29AF7FFF"),
                     facet.by = "SYMBOL",
                     xlab = "", 
                     ylab = "Counts per Million (CPM)",
                     add.params = list(size = 0.1)) +
  scale_x_discrete(labels = c("Day 0",
                              "Day 25",
                              "Day 65")) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 90))

# Place the plot  
plotGG(plot = FIPD.exp_NR2C2.FAM47E, 
       x = 6.0, 
       y = 4.75 ,
       width = 2.0, 
       height = 3.5,
       just = c("left", "top"), 
       default.units = "inches")

pageGuideHide()
dev.off()
```

### SUPPLEMENTARY FIGURE 2
IHEC consortium has one sample of substantia nigra for H3K27me3, H3K9me3, H3K4me1, H3K4me3 and H3K36me3 from a healthy brain for BAG3, IDUA and SCARB2 loci
```{r suppFig2, eval = FALSE, echo = TRUE}
############################################
########## SUPPLEMENTARY FIGURE 2 ##########
############################################
# Save as TIFF, 300 dpi
tiff("/home/vagrant/Manuscript_1/SUPPLEMENTARY_FIGURE2/SUPPLEMENTARY_FIGURE2.tiff",
     width = 8.27,
     height = 6.5,
     units = "in",
     res = 300,
     compression = "lzw")

# Create a A4 blank page
pageCreate(width = 8.27, 
           height = 6.5, 
           default.units = "inches",
           showGuides = TRUE)

#### PANEL A - 
# text Supplementary Figure 2
plotText(label = "Supplementary Figure 2", 
         fontsize = 14,
         x = 0.25, 
         y = 0.25, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")
# text A
plotText(label = "A", 
         fontsize = 12,
         x = 0.25, 
         y = 0.5, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# ChIP-seq signal BAG3
# path where bigwig files are
bw.IHEC.path = "/home/vagrant/Documents/IHEC/ChIP_Seq/BIGWIG/"

# Reads BIGWIG file at BAG3 location - H3K4me1
bw.BAG3.H3K4me1 = readBigwig(file = paste0(bw.IHEC.path,
                                             "ihec.chipseq.ihec-chipseq-containerv1.1.4.IHECRE00000986.6.6a1889bf-7312-4e41-b1cc-e2f3be7a8076.fc.signal.bigwig"),
                               chrom = "chr10",
                               chromstart = 119611305,
                               chromend = 119691505)

# Reads BIGWIG file at BAG3 location - H3K4me3
bw.BAG3.H3K4me3 = readBigwig(file = paste0(bw.IHEC.path,
                                             "ihec.chipseq.ihec-chipseq-containerv1.1.4.IHECRE00000986.6.4cb9300f-9097-4374-8f2a-4afd8ff855fb.fc.signal.bigwig"),
                               chrom = "chr10",
                               chromstart = 119611305,
                               chromend = 119691505)

# Reads BIGWIG file at BAG3 location - H3K36me3
bw.BAG3.H3K36me3 = readBigwig(file = paste0(bw.IHEC.path,
                                             "ihec.chipseq.ihec-chipseq-containerv1.1.4.IHECRE00000986.6.6586e624-852f-4cac-83df-2325cee19699.fc.signal.bigwig"),
                               chrom = "chr10",
                               chromstart = 119611305,
                               chromend = 119691505)

# Reads BIGWIG file at BAG3 location - H3K9me3
bw.BAG3.H3K9me3 = readBigwig(file = paste0(bw.IHEC.path,
                                             "ihec.chipseq.ihec-chipseq-containerv1.1.4.IHECRE00000986.6.e8fe7664-906e-4d87-8f3a-6ce88dbeadc6.fc.signal.bigwig"),
                               chrom = "chr10",
                               chromstart = 119611305,
                               chromend = 119691505)

# Reads BIGWIG file at BAG3 location - H3K27me3
bw.BAG3.H3K27me3 = readBigwig(file = paste0(bw.IHEC.path,
                                             "ihec.chipseq.ihec-chipseq-containerv1.1.4.IHECRE00000986.6.5f5024a9-2e3f-4527-82ec-0ed04bf4a848.fc.signal.bigwig"),
                               chrom = "chr10",
                               chromstart = 119611305,
                               chromend = 119691505)

# Add a scale next to the bigwig files - check the maximum to choose
scale.IHEC.BAG3.max = max(c(bw.BAG3.H3K4me1$score,
                              bw.BAG3.H3K4me3$score,
                              bw.BAG3.H3K36me3$score,
                              bw.BAG3.H3K9me3$score,
                              bw.BAG3.H3K27me3$score)) %>% 
  round(.,
        digits = 1.0)

print(scale.IHEC.BAG3.max)

# Define parameters for the regions
region.p.IHEC.BAG3 = pgParams(chrom = "chr10",
                         chromstart = 119611405,
                         chromend = 119691405,
                         assembly = "hg38",
                         range = c(0, scale.IHEC.BAG3.max))

# Add the genomic label
plotGenomeLabel(
  chrom = "chr10",
  chromstart = 119611405,
  chromend = 119691405,
  assembly = "hg38",
  x = 0.75,
  y = 0.625,
  length = 4.0,
  default.units = "inches",
  scale = "Mb")

# ChIPseq signal H3K4me1
H3K4me1_BAG3 = plotSignal(
  data = bw.BAG3.H3K4me1, 
  params = region.p.IHEC.BAG3,
  fill = "#b5de2b", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.75, 
  y = 0.875, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = H3K4me1_BAG3, 
          at = c(0, scale.IHEC.BAG3.max),
          fontsize = 6)

# ChIPseq signal H3K4me3
H3K4me3_BAG3 = plotSignal(
  data = bw.BAG3.H3K4me3, 
  params = region.p.IHEC.BAG3,
  fill = "#35b779", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.75, 
  y = 1.25, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = H3K4me3_BAG3, 
          at = c(0, scale.IHEC.BAG3.max),
          fontsize = 6)

# ChIPseq signal H3K36me3
H3K36me3_BAG3 = plotSignal(
  data = bw.BAG3.H3K36me3, 
  params = region.p.IHEC.BAG3,
  fill = "#26828e", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.75, 
  y = 1.625, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = H3K36me3_BAG3, 
          at = c(0, scale.IHEC.BAG3.max),
          fontsize = 6)

# ChIPseq signal H3K9me3
H3K9me3_BAG3 = plotSignal(
  data = bw.BAG3.H3K9me3, 
  params = region.p.IHEC.BAG3,
  fill = "#3e4989", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.75, 
  y = 2.0, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = H3K9me3_BAG3, 
          at = c(0, scale.IHEC.BAG3.max),
          fontsize = 6)

# ChIPseq signal H3K27me3
H3K27me3_BAG3 = plotSignal(
  data = bw.BAG3.H3K27me3, 
  params = region.p.IHEC.BAG3,
  fill = "#440154", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.75, 
  y = 2.375, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = H3K27me3_BAG3, 
          at = c(0, scale.IHEC.BAG3.max),
          fontsize = 6)

# Gene track
plotGenes(
  chrom = "chr10",
  chromstart = 119611405,
  chromend = 119691405,
  assembly = assembly(Genome = "hg38refGene", 
                      TxDb = "TxDb.Hsapiens.UCSC.hg38.refGene", 
                      OrgDb = "org.Hs.eg.db"),
  fontcolor = "black",
  fill = "black",
  x = 0.75,
  y = 2.75,
  width = 4.0,
  height = 0.375,
  just = c("left", "top"),
  default.units = "inches")

# Add the samples name
plotText(label = "H3K4me1", 
         fontsize = 6, 
         fontcolor = "#b5de2b",
         x = 0.75,
         y = 0.875, 
         just = c("left", "top"),
         default.units = "inches")

# Add the samples name
plotText(label = "H3K4me3", 
         fontsize = 6, 
         fontcolor = "#35b779",
         x = 0.75,
         y = 1.25, 
         just = c("left", "top"),
         default.units = "inches")

# Add the samples name
plotText(label = "H3K36me3", 
         fontsize = 6, 
         fontcolor = "#26828e",
         x = 0.75,
         y = 1.625, 
         just = c("left", "top"),
         default.units = "inches")

# Add the samples name
plotText(label = "H3K9me3", 
         fontsize = 6, 
         fontcolor = "#3e4989",
         x = 0.75,
         y = 2.0, 
         just = c("left", "top"),
         default.units = "inches")

# Add the samples name
plotText(label = "H3K27me3", 
         fontsize = 6, 
         fontcolor = "#440154",
         x = 0.75,
         y = 2.375, 
         just = c("left", "top"),
         default.units = "inches")

#### PANEL B - 
# text B
plotText(label = "B", 
         fontsize = 12,
         x = 0.25, 
         y = 3.25, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# Reads BIGWIG file at SCARB2 location - H3K4me1
bw.SCARB2.H3K4me1 = readBigwig(file = paste0(bw.IHEC.path,
                                             "ihec.chipseq.ihec-chipseq-containerv1.1.4.IHECRE00000986.6.6a1889bf-7312-4e41-b1cc-e2f3be7a8076.fc.signal.bigwig"),
                               chrom = "chr4",
                               chromstart = 76173617,
                               chromend = 76253817)

# Reads BIGWIG file at SCARB2 location - H3K4me3
bw.SCARB2.H3K4me3 = readBigwig(file = paste0(bw.IHEC.path,
                                             "ihec.chipseq.ihec-chipseq-containerv1.1.4.IHECRE00000986.6.4cb9300f-9097-4374-8f2a-4afd8ff855fb.fc.signal.bigwig"),
                               chrom = "chr4",
                               chromstart = 76173617,
                               chromend = 76253817)

# Reads BIGWIG file at SCARB2 location - H3K36me3
bw.SCARB2.H3K36me3 = readBigwig(file = paste0(bw.IHEC.path,
                                             "ihec.chipseq.ihec-chipseq-containerv1.1.4.IHECRE00000986.6.6586e624-852f-4cac-83df-2325cee19699.fc.signal.bigwig"),
                               chrom = "chr4",
                               chromstart = 76173617,
                               chromend = 76253817)

# Reads BIGWIG file at SCARB2 location - H3K9me3
bw.SCARB2.H3K9me3 = readBigwig(file = paste0(bw.IHEC.path,
                                             "ihec.chipseq.ihec-chipseq-containerv1.1.4.IHECRE00000986.6.e8fe7664-906e-4d87-8f3a-6ce88dbeadc6.fc.signal.bigwig"),
                               chrom = "chr4",
                               chromstart = 76173617,
                               chromend = 76253817)

# Reads BIGWIG file at SCARB2 location - H3K27me3
bw.SCARB2.H3K27me3 = readBigwig(file = paste0(bw.IHEC.path,
                                             "ihec.chipseq.ihec-chipseq-containerv1.1.4.IHECRE00000986.6.5f5024a9-2e3f-4527-82ec-0ed04bf4a848.fc.signal.bigwig"),
                               chrom = "chr4",
                               chromstart = 76173617,
                               chromend = 76253817)

# Add a scale next to the bigwig files - check the maximum to choose
scale.IHEC.SCARB2.max = max(c(bw.SCARB2.H3K4me1$score,
                              bw.SCARB2.H3K4me3$score,
                              bw.SCARB2.H3K36me3$score,
                              bw.SCARB2.H3K9me3$score,
                              bw.SCARB2.H3K27me3$score)) %>% 
  round(.,
        digits = 1.0)

print(scale.IHEC.SCARB2.max)

# Define parameters for the regions
region.p.IHEC.SCARB2 = pgParams(chrom = "chr4",
                         chromstart = 76173717,
                         chromend = 76253717,
                         assembly = "hg38",
                         range = c(0, scale.IHEC.SCARB2.max))

# Add the genomic label
plotGenomeLabel(
  chrom = "chr4",
  chromstart = 76173717,
  chromend = 76253717,
  assembly = "hg38",
  x = 0.75,
  y = 3.5,
  length = 4.0,
  default.units = "inches",
  scale = "Mb")

# ChIPseq signal H3K4me1
H3K4me1_SCARB2 = plotSignal(
  data = bw.SCARB2.H3K4me1, 
  params = region.p.IHEC.SCARB2,
  fill = "#b5de2b", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.75, 
  y = 3.75, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = H3K4me1_SCARB2, 
          at = c(0, scale.IHEC.SCARB2.max),
          fontsize = 6)

# ChIPseq signal H3K4me3
H3K4me3_SCARB2 = plotSignal(
  data = bw.SCARB2.H3K4me3, 
  params = region.p.IHEC.SCARB2,
  fill = "#35b779", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.75, 
  y = 4.125, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = H3K4me3_SCARB2, 
          at = c(0, scale.IHEC.SCARB2.max),
          fontsize = 6)

# ChIPseq signal H3K36me3
H3K36me3_SCARB2 = plotSignal(
  data = bw.SCARB2.H3K36me3, 
  params = region.p.IHEC.SCARB2,
  fill = "#26828e", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.75, 
  y = 4.5, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = H3K36me3_SCARB2, 
          at = c(0, scale.IHEC.SCARB2.max),
          fontsize = 6)

# ChIPseq signal H3K9me3
H3K9me3_SCARB2 = plotSignal(
  data = bw.SCARB2.H3K9me3, 
  params = region.p.IHEC.SCARB2,
  fill = "#3e4989", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.75, 
  y = 4.875, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = H3K9me3_SCARB2, 
          at = c(0, scale.IHEC.SCARB2.max),
          fontsize = 6)

# ChIPseq signal H3K27me3
H3K27me3_SCARB2 = plotSignal(
  data = bw.SCARB2.H3K27me3, 
  params = region.p.IHEC.SCARB2,
  fill = "#440154", 
  alpha = 0.7, 
  linecolor = NA,
  x = 0.75, 
  y = 5.25, 
  width = 4.0, 
  height = 0.25,
  just = c("left", "top"), 
  default.units = "inches")

annoYaxis(plot = H3K27me3_SCARB2, 
          at = c(0, scale.IHEC.SCARB2.max),
          fontsize = 6)

# Gene track
plotGenes(
  chrom = "chr4",
  chromstart = 76173717,
  chromend = 76253717,
  assembly = assembly(Genome = "hg38refGene", 
                      TxDb = "TxDb.Hsapiens.UCSC.hg38.refGene", 
                      OrgDb = "org.Hs.eg.db"),
  fontcolor = "black",
  fill = "black",
  x = 0.75,
  y = 5.625,
  width = 4.0,
  height = 0.375,
  just = c("left", "top"),
  default.units = "inches")

# Add the samples name
plotText(label = "H3K4me1", 
         fontsize = 6, 
         fontcolor = "#b5de2b",
         x = 0.75,
         y = 3.75, 
         just = c("left", "top"),
         default.units = "inches")

# Add the samples name
plotText(label = "H3K4me3", 
         fontsize = 6, 
         fontcolor = "#35b779",
         x = 0.75,
         y = 4.125, 
         just = c("left", "top"),
         default.units = "inches")

# Add the samples name
plotText(label = "H3K36me3", 
         fontsize = 6, 
         fontcolor = "#26828e",
         x = 0.75,
         y = 4.5, 
         just = c("left", "top"),
         default.units = "inches")

# Add the samples name
plotText(label = "H3K9me3", 
         fontsize = 6, 
         fontcolor = "#3e4989",
         x = 0.75,
         y = 4.875, 
         just = c("left", "top"),
         default.units = "inches")

# Add the samples name
plotText(label = "H3K27me3", 
         fontsize = 6, 
         fontcolor = "#440154",
         x = 0.75,
         y = 5.25, 
         just = c("left", "top"),
         default.units = "inches")
  
pageGuideHide()  
dev.off()
```

Synchronise the files between my personal ATLAS and the FSTC.SYSBIO ATLAS
```{bash, eval = FALSE, echo = TRUE}
cd /Volumes/FSTC_SYSBIO/13-\ PAPERS/2025_Gerard_et_al_SCARB2
rsync -avzPhu --no-p --chmod=ugo=rwx --delete /Volumes/deborah.gerard/Documents/Manuscript_1/ .
```

### SUPPLEMENTARY FIGURE 3
Plot FACS results for experiments where knockdown were performed at day 9 and samples collected 3 and 6 days post-transduction
```{r suupFig3, eval = FALSE, echo = TRUE}
############################################
########## SUPPLEMENTARY FIGURE 3 ##########
############################################
# Save as a PDF
pdf("/home/vagrant/Manuscript_1/SUPPLEMENTARY_FIGURE3/SUPPLEMENTARY_FIGURE3.pdf",
    width = 13, 
    height = 15)

# Save as TIFF, 300 ppi
tiff("/home/vagrant/Manuscript_1/SUPPLEMENTARY_FIGURE3/SUPPLEMENTARY_FIGURE3.tiff",
     width = 8.27,
     height = 11.67,
     units = "in",
     res = 300,
     compression = "lzw")

# Create a A4 blank page
pageCreate(width = 8.27, 
           height = 11.67, 
           default.units = "inches",
           showGuides = TRUE)

#### PANEL A - 
# text Supplementary Figure 3
plotText(label = "Supplementary Figure 3", 
         fontsize = 14,
         x = 0.25, 
         y = 0.25, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")
# text A
plotText(label = "A", 
         fontsize = 12,
         x = 0.25, 
         y = 0.5, 
         just = "left", 
         default.units = "inches",
         fontface = "bold")

# Read the facs files (.fcs)
#facs.path = "/home/vagrant/Documents/Wetlab/epifunc/"
facs.path = "/Volumes/deborah.gerard/Documents/Wetlab/epifunc/"

facs_6days_PT_N1.fs = read.flowSet(path = paste0(facs.path,
                                              "2022.11_DG_KD_SCARB2_NR2C2_1/FACS/RAW_DATA/NEW_EXPORT_06122022_1624/2022.11_DG_KD_SCARB2_NR2C2_1/"),
                                pattern = ".fcs",
                               alter.names = TRUE,
                               truncate_max_range = FALSE)

# Check what channels are present
colnames(facs_6days_PT_N1.fs)

# Check sample
pData(facs_6days_PT_N1.fs)

# Rename some for clarity
colnames(facs_6days_PT_N1.fs)[colnames(facs_6days_PT_N1.fs) == "FITC.A"] = "GFP"
colnames(facs_6days_PT_N1.fs)[colnames(facs_6days_PT_N1.fs) == "mCherry..A"] = "mCHERRY"
colnames(facs_6days_PT_N1.fs)[colnames(facs_6days_PT_N1.fs) == "SYTOX_BLUE..A"] = "SYTOX BLUE"

# To be able to add gates, the object must be transformed as a GatingSet object
facs_6days_PT_N1.gs = GatingSet(facs_6days_PT_N1.fs)

# Define neurons gate
dead.neur.gate = polygonGate(filterId = "Neurons",
                        "FSC.A" = c(5e4, 2e5, 2e5, 5e4),
                        "SSC.A" = c(0e4, 125e3, 2.e5, 1e5))

# Plot SYTOX BLUE positive CTRL
ggcyto(facs_6days_PT_N1.gs[[1]],
       aes(x = FSC.A, 
           y = SSC.A),
       subset = "root") + 
  geom_hex(bins = 200) +
  geom_gate(dead.neur.gate) +
  ggcyto_par_set(limits = "instrument")

# The gate for dead neurons has been defined -> add the gate to the gateset and recompute
add(facs_6days_PT_N1.gs,
    dead.neur.gate)

recompute(facs_6days_PT_N1.gs)

dev.off()
```

SCARB2 snp (rs1465922) for samples with whole genome sequencing data (75 cases)
```{r, eval= FALSE, echo = TRUE}
# Load data
SCARB2.SNP.WGS.75 = read_excel("/Volumes/deborah.gerard/Documents/epifunc/Final.WGS.SCARB2.OK_from_Sinthu.xlsx")

# Filter for rs1465922
SCARB2.SNP.WGS.75 %>% 
  filter(avsnp150 == "rs1465922") %>% 
  dplyr::select(samples) %>% 
  separate(samples,
           sep = ", ",
           into = )
  
```


```{r, eval = FALSE, echo = TRUE}
sessionInfo()
```

```{r, eval = FALSE, echo = TRUE}
library("tidyverse")
library("readxl")
library("AnnotationDbi")
library("EnsDb.Hsapiens.v86")

# Load data
TFlist = read_excel("/Volumes/deborah.gerard/Documents/Elena/Human_TF_list.xls",
                    skip = 1,
                    col_names = TRUE)

# Add gene symbol
TFlist.GN = select(EnsDb.Hsapiens.v86, 
       keys = as.character(TFlist$EntrezID), 
       columns = c("ENTREZID", "GENENAME"), 
       keytype = "ENTREZID")

TFlist.GN %>% 
  as_tibble() %>% 
  distinct(GENENAME)
  write_delim(.,
              "/Volumes/deborah.gerard/Documents/Elena/Human_TF_list_geneName.txt",
              col_names = TRUE,
              delim = "\t")

```

